\documentclass{article}
\title{Notes on evolution of sexual reproduction}
\usepackage{amsmath}
\usepackage{natbib}
\usepackage{hyperref}
\bibliographystyle{chicago}
\date{\today}

\begin{document}

\maketitle

\section{Introduction (mostly based on \cite{otto2009evolutionary})}

\cite{hartfield2012current} also provides a good review.

Also see \cite{lehtonen2012many} for review on cost of sex.

We want to understand the evolution of sexual reproduction and model the freshwater snail system studied by \cite{vergara2014infection}. \cite{otto2009evolutionary} provides a fairly recent review in this field.

There are a few disadvantages to sexual reproduction \citep{otto2009evolutionary}:

\begin{itemize}
    \item partner searching, resource gathering, transmission of STD, and predation while mating.
    \item Only 50\% of the parent gene is passed down to its offspring (often referred to as the cost of sex \citep{bell1982masterpiece}).
    \item Recombination may alter favourable gene combination \citep{nei1967modification}.
\end{itemize}

 
Despite such costs, sexual reproduction is a widespread phenomenon among eukaryotic organisms.
One of the oldest explanation for sex is that it increases variation \citep{weissmann1889significance} but this is a misconception \citep{otto2009evolutionary}. 
Not only does sexual reproduction can decrease varation but varation generated through sex can also reduce fitnesss (segration load).
So why does sex exist?

\subsection{Earlier models}

\begin{itemize}
    \item Focused on modifier genes but found that selections acts againt sex and recombination at equilibrium \citep{feldman1996population}. This idea applies even when sexual reproduction does not have any cost. Maybe it's important to consider nonequilibrium situations!
    \item In an evolving population, modifiers increasing levels of recombination may be selected if "fitness surfaces were negatively curved on a multiplicative scale" \citep{barton1995general, charlesworth1990mutation}. However, these models are limited because negative fitness surface imposes fitness load. Moderately negative fitness curvature is necessary \citep{otto2003advantages}.
    \item Evidence is not consistent with this idea and the allowed curvature parameter is too narrow to be likely.
\end{itemize}

\subsection{Recent models}

\subsubsection*{Varying selection over time (Red Queen)}

\begin{itemize}
    \item In a rapidly changing environment, segregation and recombination can increase fitness of near descendants by dissociating detrimention genetic association \citep{salathe2008rapid}.
    \item Under antogonistic interaction, species must evolve fast enough so that remain in pace with the interacting species (Red Queen) \citep{bell1982masterpiece}.
    \item Limitations: (1) selection must be quite strong in haploid populations; (2) In diploid populations, where form of selection does not cycle, genetic association built up by past selection never becomes detrimental and segregation does not provide any benefit. See \cite{otto2009evolutionary} for more sources.
    \item Red queen is still beneficial when (1) parasites originate from close relatives and (2) it is "a source of ongoing selection". See \cite{otto2009evolutionary} for more sources.
\end{itemize}

\subsubsection*{Varying selection over space}

\begin{itemize}
    \item In a spatially heterogeneous environment, recombination can mitigate detrimental genetric association from migration. See \cite{otto2009evolutionary} for more sources.
    \item See Salathe et al 2006 and Martin et al 2006
\end{itemize}

\subsubsection*{Rates of sex vary among individuals}

\begin{itemize}
    \item Many organisms are more likely to engage in sex when they are in poor condition (e.g. Daphnia). See \cite{otto2009evolutionary} for more sources.
    \item "Models that have investigated the evolution
of condition-dependent sex have found it much easier for
sex to evolve if individuals in worse condition allocate
more resources to sexual reproduction than do individuals
in good condition" \citep{otto2009evolutionary}.
    \item "Ineffective at selecting for recombination in diploid individuals" \citep{agrawal2005evolution}.
\end{itemize}

\subsection*{Selection in finite population}

\begin{itemize}
    \item In an infinite population, recombination is not required to combine advantageous mutations. This is not the case in a finite population. See \cite{otto2009evolutionary} for more sources.
    \item Muller's ratchet \citep{muller1964relation}.
    \item In asexual populations, genome containing both advantagenous mutants would occur on the order of $1/\sqrt{N\mu^2}$ generations, whereas sexual population takes $1/(RN\mu^2/3)^{1/3}$ order of generations \citep{christiansen1998waiting}.
    \item Hill-Robertson effect. See \cite{hartfield2012current} for more sources.
    \item (1) Results are less sensitive to epistasis; (2) population size does not need to be too small; and (3) can explain combination of selecion (e.g. deleterious mutations and the Red Queen \citep{howard1994parasitism}).
    \item Not very clear whether this explanation is strong enough.
\end{itemize}

\section{Red Queen}

See \cite{hartfield2012current} for a review of the history of the Red Queen hypothesis.

See Gandon and Otto 2007 and Barton 2010....

Modifiers spread due to delayed short-term benefit

Recent paper by \cite{slowinski2016coevolutionary} provides an experimental work on RQH. Study is based on three treatments: avirulent pathogen treatment, fixed pathogen treatment, and coevolving pathogen treatment. Asexual failed to invade the last group but managed to persist with a low frequency in some population (See \citep{agrawal2001parasites} for the theoretical model).

\cite{agrawal2001parasites} present four allele matching host-parasite models that can be used for our models. They do not include gene-for-gene interaction models (see cited papers for further detail). They assume infinite size so we should consider finite size model.

See \cite{otto2004species} for broader model.

See Vergara et al 2014 for periodicity...

See \cite{ashby2015diversity} for a stochastic model on RQH. They find that sex is more likely to persist when there is more genetic diversity in the population only if transmission rate is high enough. If transmission rate is low, high genetic diversity selects against sex. Their result may seem to contrast that of \cite{lively2010epidemiological} because their model allows for the effects of drift but it is actually consistent. This paper also summarizes all the other model studies, which might be helpful. See supplementary materials for the code.

See \cite{ashby2014parasitic} for a deterministic model. Also see \cite{may1983epidemiology} for an old model.

\section{Snails}

There's a lot going on in the snails population. Here are some things we want to consider for the model:

\begin{itemize}
    \item Some old papers on snails: \citep{lively1987evidence; lively1989adaptation}
    \item See \cite{dybdahl1998host} for time-lagged selection.
    \item Spatial structure within/among lakes (geographic mosaic theory) \citep{king2009geographic}. There's also fine scale spatial heterogeneous prevalence around the lake, more than one-third of which can be explained by variation in mean susceptibility (we certainly need to use a model with more than two loci) \citep{gibson2016fine}. See \cite{mckone2016fine} for red-queen related stuff.
    \item Genetic diversity among different lakes - multiple loci is probably necessary to produce enough genetic diversity 
    \item Life cycle of the parasite. Do we need to model ducks as well? We might also have to model the sexual reproduction of the parasites \citep{hechinger2012faunal}
    \item Finite population (stochastic) vs. infinite population (deterministic)
    \item Continuous time vs. discrete time (See \cite{lively2010epidemiological})
    \item Cost of sex. A very recent paper by \cite{gibson2017two} show that snail population support two-fold cost of sex (or slightly greater based on MLE). They also found that asexual females produce more offspring than sexual females.
\end{itemize}

\section{Model}

\cite{dybdahl1998host} provides a model for asexual snails and sexual parasite but is too simple...

Let's model sexual and asexual individuals in the absence of parasites first. We're going to use two alleles for each of the two loci. Then, there are four types of gametes that can be produced: $AB, Ab, aB, ab$. 
We will be using indices $1-4$ for simplicity \citep{agrawal2006host}.

The number of sexual individuals with genotype $ij$ is given by $S_{ij}$. Then, we can model $S_{ij}$ in the next generation as follows:
$$
\begin{aligned}
S_{ii}' &= F_0 (M_i F_i) W_U\\
S_{ij}' &= F_0 (M_i F_j + M_j F_i) W_U,
\end{aligned}
$$
where $F_0 =$ is the number of females $M_i$ and $F_i$ are proportion of male and female gametes $i$. Here, $F_0$ is defined as $(1-s) S_0$ where $s$ is the proportion of males produced and $S_0$ is the number of sexuals in the population.
In a deterministic model, since $M_i = F_i$, we get replace it with $g_i$, the proportion of the gamete $i$ in the sexual population:
$$
\begin{aligned}
S_{ii}' &= (1-s) S_0 (g_i^2) W_U\\
S_{ij}' &= (1-s) S_0 (2 g_i g_j) W_U.
\end{aligned}
$$


$$
\begin{aligned}
g_1 &= (\sum_{k} S_{1k} - r S_{14} + r S_{23})/(\sum_j X_j),\\
g_2 &= (\sum_{k} S_{2k} - r S_{23} + r S_{14})//(\sum_j X_j),\\
g_3 &= (\sum_{k} S_{3k} - r S_{23} + r S_{14})//(\sum_j X_j),\\
g_4 &= (\sum_{k} S_{4k} - r S_{14} + r S_{23})//(\sum_j X_j).\\
\end{aligned}
$$

Clonal asexuals are easy to model:
$$
A_{ij}' = A_{ij} W_U
$$
We follow \cite{lively2010epidemiological} and define
$$
W_U = \frac{b_U}{1 + (a_U N)^x}.
$$

<<discrete1>>==
discrete_par <- c(
    s=0.5,
    r=0.2,
    aU=0.001,
    bU=10,
    tmax=100,
    N0=100
)

simulate_noinfection <- function(param){
    with(as.list(param), {
        ##TODO: write initializing function
        S <- A <- array(0, dim=c(4, 4, tmax))
        S.count <- rep(0,tmax)
        A.count <- rep(0,tmax)
        S0 <- matrix(runif(16), 4, 4)
        A0 <- matrix(runif(16), 4, 4)
        upr <- upper.tri(S0, diag=TRUE)
        
        A[,,1] <- A0 + t(A0)
        A[,,1] <- N0*A[,,1]/sum(A[,,1][upr])
        S[,,1] <- S0 + t(S0)
        S[,,1] <- N0*S[,,1]/sum(S[,,1][upr])
        
        S.count[1] <- N0
        A.count[1] <- N0
        
        for(t in 1:(tmax-1)){
            N <- sum(S[,,t][upr]) + sum(A[,,t][upr])
            WU <- bU/(1+aU*N)
            
            recomb.gamete <- outer(c(-1, 1, 1, -1), c(S[1,4,t],-S[2,3,t]), "*")
            S.gamete <- colSums(S[,,t]) + r * rowSums(recomb.gamete)
            if(sum(S.gamete)!=0) {
                S.gamete <- S.gamete/sum(S.gamete)
            }
            
            S.outcross <- outer(S.gamete, S.gamete, "*")
            S.outcross <- 2*S.outcross
            diag(S.outcross) <- diag(S.outcross)/2
            S[,,t+1] <- (1-s) * S.count[t] * S.outcross * WU
            S.count[t+1] <- sum(S[,,t+1][upr])
            
            A[,,t+1] <- A[,,t] * WU
            A.count[t+1] <- sum(A[,,t+1][upr])
            
        }
        
        return(list(
            S=S, S.count=S.count, A=A, A.count= A.count
        ))
    })
}

res_noinfection <- simulate_noinfection(discrete_par)
plot(res_noinfection$A.count, type="l")
lines(res_noinfection$S.count, col=2)
@

OK, let's try to add infection now.

Following \cite{lively2010epidemiological}, we can write
$$
\begin{aligned}
S_{ii}' &= (1-s) S_0 (g_i^2) [W_I P_{ij} + W_U(1-P_{ij})]\\
S_{ij}' &= (1-s) S_0 (2 g_i g_j) [W_I P_{ij} + W_U(1-P_{ij})]\\
A_{ij}' &= A_{ij} [W_I P_{ij} + W_U (1-P_{ij})]
\end{aligned}
$$
Without sexual reproduction of parasites, $P_{ij}$ is defined in the following way:
$$
P_{ij} = 1-\exp(-\beta[S_{ij,I} + A_{ij,I}]/N')
$$

To acconut for sexual reproduction of parasites, we have to adjust the equation above. Let $I = S_{\cdot, I} + A_{\cdot, I}$ represent the number of infected hosts. Then, we can define $\rho_{ij} = [S_{ij,I} + A_{ij,I}]/I$ as the proportion of parasites with genotype $ij$. Then, we can define the proportion of gametes in the parasite population, $h_{i}$, as we did above. Then,
$$
\begin{aligned}
\rho_{ii}' &= h_i^2\\
\rho_{ij}' &= 2 h_i h_j
\end{aligned}
$$
Then, $P_{ij}$ accounting for recombination is
$$
P_{ij} = 1-\exp(-\beta[\rho_{ij}' I]/N')
$$

<<>>==
discrete_par2 <- c(
    s=0.5,
    r=0.2,
    beta=20,
    aI=0.001,
    aU=0.001,
    bU=20,
    bI=2,
    tmax=2100,
    tburnin=1000,
    N0=100,
    I0=1
)

simulate_infection <- function(param){
    with(as.list(param), {
        ##TODO: write initializing function
        S <- SI <- A <- AI <- array(0, dim=c(4, 4, tmax))
        N.count <- S.count <- SI.count <- A.count <- AI.count <- rep(0,tmax)
        S0 <- matrix(runif(16), 4, 4)
        upr <- upper.tri(S0, diag=TRUE)
        
        S[,,1] <- S0 + t(S0)
        S[,,1] <- N0*S[,,1]/sum(S[,,1][upr])
        SI[,,1] <- I0*S[,,1]/sum(S[,,1][upr])
        
        S.count[1] <- N0
        SI.count[1] <- I0
        N.count[1] <- S.count[1] + A.count[1] + SI.count[1] + AI.count[1]
        
        I <- sum((AI[,,1]+SI[,,1])[upr])
        rho <- (SI[,,1] + AI[,,1])/I
        rho.recomb <- outer(c(-1, 1, 1, -1), c(rho[1, 4], -rho[2,3]), "*")
        rho.gamete <- colSums(rho) + r * rowSums(rho.recomb)
        rho.gamete <- rho.gamete/sum(rho.gamete)
        para.outcross <- outer(rho.gamete, rho.gamete, "*")
        para.outcross <- 2*para.outcross
        diag(para.outcross) <- diag(para.outcross)/2
        P <- 1 - exp(-beta*para.outcross*I/N.count[1])
        
        for(t in 1:(tmax-1)){
            if (t==tburnin) {
                ## FIXME: This doesn't give uniform probability
                A0 <- matrix(as.numeric(1:16 == ceiling(runif(1, 0, 16))), 4, 4)
                A0 <- A0 + t(A0)
                diag(A0) <- diag(A0)/2
                A[,,t] <- A0
                A.count[t] <- 1 
            }
            
            WU <- bU/(1+aU*N.count[t])
            WI <- bI/(1+aI*N.count[t])
            
            Sp <- (1-s) * S[,,t] * (WI * P + WU * (1-P))
            S.count[t+1] <- sum(Sp[upr])
            recomb.gamete <- outer(c(-1, 1, 1, -1), c(Sp[1,4],-Sp[2,3]), "*")
            S.gamete <- colSums(Sp) + r * rowSums(recomb.gamete)
            if(sum(S.gamete)!=0) {
                S.gamete <- S.gamete/sum(S.gamete)
            }
            
            S.outcross <- outer(S.gamete, S.gamete, "*")
            S.outcross <- 2*S.outcross
            diag(S.outcross) <- diag(S.outcross)/2
            
            S[,,t+1] <- S.count[t+1] * S.outcross
            
            A[,,t+1] <- A[,,t] * (WI * P + WU * (1-P))
            A.count[t+1] <- sum(A[,,t+1][upr])
            AI[,,t+1] <- A[,,t] * P
            AI.count[t+1] <- sum(AI[,,t+1][upr])
            
            SI[,,t+1] <- S[,,t] * P
            SI.count[t+1] <- sum(SI[,,t+1][upr])
            
            N.count[t+1] <- S.count[t+1] + A.count[t+1] + SI.count[t+1] + AI.count[t+1]
            
            I <- sum((AI[,,t+1]+SI[,,t+1])[upr])
            rho <- (SI[,,t+1] + AI[,,t+1])/I
            rho.recomb <- outer(c(-1, 1, 1, -1), c(rho[1, 4], -rho[2,3]), "*")
            rho.gamete <- colSums(rho) + r * rowSums(rho.recomb)
            rho.gamete <- rho.gamete/sum(rho.gamete)
            para.outcross <- outer(rho.gamete, rho.gamete, "*")
            para.outcross <- 2*para.outcross
            diag(para.outcross) <- diag(para.outcross)/2
            P <- 1 - exp(-beta*para.outcross*I/N.count[t+1])
        }
        
        return(list(
            S=S, SI=SI, S.count=S.count, SI.count=SI.count, A=A, AI=AI, A.count=A.count, AI.count=AI.count
        ))
    })
}

##TODO: Add a bit of stochasticity to prevent fixation

set.seed(101)
infection_res <- simulate_infection(discrete_par2)
tframe <- 950:1200
## sex vs asex
plot(tframe, infection_res$S.count[tframe], type="l", ylim=c(0,1e4), main="sex vs. asex size", ylab="size", xlab="generation")
lines(tframe, infection_res$A.count[tframe], col=2)

## sex vs asex prevalence
with(infection_res,{
    plot(tframe, SI.count[tframe]/(S.count[tframe]+SI.count[tframe]), type="l", ylim=c(0,1), main="sex vs. asex prevalence", ylab="prop")
    lines(tframe, AI.count[tframe]/(A.count[tframe]+AI.count[tframe]), col=2)
})

@

\bibliography{redqueen}
\end{document}
