\documentclass{article}
\title{Quantifying the effects of the Red Queen}

\usepackage{tabularx}

\usepackage{amsmath}
\usepackage{natbib}
\usepackage{hyperref}
\bibliographystyle{chicago}
\date{\today}

\begin{document}

\maketitle

\section{Introduction}

Sexually reproducing organisms suffer from inevitable costs \citep{lehtonen2012many}.
These costs are often attributed to cost of producing males \citep{smith1978evolution}.
As males cannot produce offspring, a sexually reproducing population should be outgrown by an asexually reproducing population growing twice as fast (given that the sexual population produces 50\% males and 50\% females); hence the infamous \emph{two-fold cost of sex} \citep{smith1978evolution}.
However, the two-fold cost of sex relies on the assumption that all else is equal, and this is not necessarily true in nature.
In fact, sexual reproduction is much more common in nature \citep{vrijenhoek1998animal}.
So what drives sex to persist?

One explanation for sex is the Red Queen Hypothesis \citep{bell1982masterpiece}.
The Red Queen Hypothesis predicts a sexual population to overcome the two-fold cost of sex under strong parasite selection by producing genetically diverse offspring that are resistant to infection \citep{jbs1949disease, jaenike1978hypothesis, hamilton1980sex}.
As a result, hosts with rare genotypes are expected to result in higher fitness and be common eventually.
Parasites coevolves with the host population to infect the common "rare" genotype [CITE],
and negative frequeny dependnece drives an oscillation in genotypic frequencies in host and parasite populations \citep{clarke1976ecological, hamilton1980sex}.

Since the advent of the Red Queen Hypothesis, much of theoretical work has focused on figuring out conditions under which parasites can maintain sexual reproduction.

On the other hand, empiricists have tried to quantify the Red Queen. 

While the Red Queen Hypothesis is well supported both empirically [CITE] and theoretially [CITE], there still remains a gap between the evolutionary theory and the ecological data [CITE?]. Many models for Red Queen Hypothesis rely on simple assumption that are not applicable to natural populations \citep{ashby2015diversity}.

Recently, growing number of studies have noted the importance of ecological feedbacks on maintenance of sex [CITE].

\cite{lively2001trematode} successfully applied ???. Using a mathematical model, he predicted that prevalence of infection should be positively correlated with prevalence of sex given large variaton among population \citep{lively2001trematode} and was able to confirm empirically [CITE]. However, such correlation was not observed in a similar snail-trematode system [CITE] and Lively's original model rather relies on simple assumption.

Here, we try to bridge the gap between theory and data further by fitting a dynamical model to data. 



In this study, we show that . Then, we perform power analysis to test how likely it is to detect the positive correlation predicted by \cite{lively2001trematode}. We show that testing for positive correlation under(?)s the effect of oscillation and lack biological interpretation.

\section{Methods}

\subsection{Data}

We consider two snail-trematode population in New Zealand \citep{mckone2016fine, vergara2014infection} and a similar snail-trematode population in Isral \citep{dagan2013clonal}. First, the snail-trematode system in New Zealand has been extensively studied under the Red Queen Hypothesis [CITE] so a Red Queen model should fit well. Second, [Something about Israel snails being different]. Finally, by considering both spatial \citep{dagan2013clonal, mckone2016fine} and spatiotemporal data \citep{vergara2014infection}, we can compare how structure of the data may affect model fits. 
Data for \cite{vergara2014infection} and \cite{dagan2013clonal} were obtained from their Dryad repository [CITE] and data for \cite{mckone2016fine} was extracted from their figure.

\subsection{Model}

We model obligate sexual hosts competing with obligate asexual hosts in multiple population by extending the model introduced by \cite{lively2010epidemiological}.
A model that \cite{lively2010epidemiological} used is essentially an SI model on a discrete time scale with natural mortality and virulence (reduction in offspring production by infected hosts).
It is a suitable canditate for this study as it captures essential structures that are present in basic epidemiological and population population dynamics models and is general enough to be applied to broad systems in nature.
We do not consider mechanistic details of the snail-trematode system such as life history of trematodes \citep{vergara2014infection} as it would require us to estimate or make assumptions on parameters on mechanisms, both of which are unrealistic.

All hosts are assumed to be diploids with two biallelic loci, and parasites are assumed to be haploids.
Let $S_{ij}^k(t)$ and $A_{ij}^k(t)$ be the number of sexual and asexual hosts with genotype $ij$ from population $k$ at generation $t$. 
For simplicity, we drop the superscript representing population and write $S_{ij}(t)$ and $A_{ij}(t)$ unless necessary.
Following \cite{lively2010epidemiological}, the expected number of offsprings (without recombination or outcrossing) produced by a sexual population is given by
\begin{equation}
S_{ij}' = c_b (1-s) \left(W_U S_{ij,U} (t) + W_I S_{ij,I} (t)\right),
\end{equation}
where $s$ is the proportion of males produced, $S_{ij, U}$ and $S_{ij,I}$ are the number of uninfected and infected sexual hosts in a population, and $W_U$ and $W_I$ are their corresponding fitnesses.
We allow for cost of sex to vary by multiplying $c_b$ to the growth term ($2/c_b$ corresponds to two fold cost of sex) \citep{ashby2015diversity}.
Equivalently, $S_{ij}'$ is the amount of genotypic contribution by sexual hosts to the next generation.
We denote the expected number of offspring produced by a sexual population after accounting for recombintation and outcrossing by $S_{ij}''$.

Asexual hosts are assumed to be strictly clonal. Then, the expected number of offsprings produced by an asexual population is given by
\begin{equation}
A_{ij}' = W_U A_{ij,U} (t) + W_I A_{ij,I} (t),
\end{equation}
where $A_{ij, U}$ and $A_{ij,I}$ are the number of uninfected and infected asexual hosts in a population.

We assume that there is mixing between population, determined by rate $\epsilon_{\textrm{mix}}$.
Then, the expected number of offspring in the next generation (accounting for contributions from all populations) is given by
\begin{equation}
\begin{aligned}
\mathrm{E}\left(S_{ij}^k(t+1)\right) &= (1 - \epsilon_{\textrm{mix}}) \left(S_{ij}^k\right)'' + \frac{\epsilon_{\textrm{mix}}}{n_{\textrm{population}}-1} \sum_{l \neq k} \left(S_{ij}^l\right)'',\\
\mathrm{E}\left(A_{ij}^k(t+1)\right) &= (1 - \epsilon_{\textrm{mix}}) \left(A_{ij}^k\right)' + \frac{\epsilon_{\textrm{mix}}}{n_{\textrm{population}}-1} \sum_{l \neq k} \left(A_{ij}^l\right)',\\
\end{aligned}
\end{equation}
where $n_{\textrm{population}}$ is the number of populations present in the system.
We then take a poisson random variable to simulate process error and allow for stochastic migration to avoid fixation:
\begin{equation}
\begin{aligned}
S_{ij}^k(t+1) &\sim \mathrm{Poisson}\left(\lambda=\mathrm{E}\left(S_{ij}^k(t+1)\right)\right) + \mathrm{Bernoulli}\left(p=p_{\textrm{sex}}\right),\\
A_{ij}^k(t+1) &\sim \mathrm{Poisson}\left(\lambda=\mathrm{E}\left(A_{ij}^k(t+1)\right)\right) + \mathrm{Bernoulli}\left(p=p_{\textrm{asex}}\right),
\end{aligned}
\end{equation}
where $p_\textrm{sex}$ and $p_\textrm{asex}$ are the probability of sexual and asexual host entering the popualation.

Infection is modeled using a matching alleles model \citep{otto1998evolution}. 
Using different genetic models can lead to various dynamics \citep{agrawal2002infection},
but 


We assume that heterozygous individuals are 

To model infection, we combine the approaches by \cite{lively2010epidemiological} and \cite{ashby2015diversity}.
The expected number of infected hosts that carry parasite with genotype $i$ at generation $t$ is given by:
\begin{equation}
I_{i}(t) = \sum_{p} 2^{\delta_{ij}} \left( S_{ip,i,I}(t) + A_{ip,i,I}(t)\right),
\end{equation}
where $S_{ip,i,I}$ and $A_{ip,i,I}$ are the expected number of sexual and asexual hosts infected with genotype $i$ parasite. 
Following \cite{ashby2015diversity}, we assume that mutation can occur in one locus with probability $\epsilon$. 
We also allow for external migration with probability $p_\textrm{parasite}$ to avoid fixation.

Let $\lambda_i$ be the poisson mean number of exposures caused by infected hosts from a population in next generation \cite{lively2010epidemiological}:
\begin{equation}
\lambda_i = \frac{\beta}{2 N(t+1)} I_i',
\end{equation}
where $\beta^k$ represents parasite fecundity at site $k$ \cite{lively2010epidemiological}.
Assuming that infected hosts can also mix with hosts in other sites, the mean number of exposures to parasites with genotype $i$ that a susceptible host in site $k$ receives is
\begin{equation}
\lambda_{i, \textrm{total}}^k = (1 - \epsilon_{\textrm{site}}) \lambda_i^k  + \frac{\epsilon_{\textrm{site}}}{n_\textrm{site}-1} \sum_{l \neq k} \lambda_i^l. 
\end{equation}
Then, a force of infection from parasite $i$ that a susceptible host with genotype $ij$ in site $k$ experiences is
\begin{equation}
\mathrm{FOI}_{ij}^k = \lambda_{i, \textrm{total}}^k + \lambda_{j, \textrm{total}}^k.
\end{equation}
The probability of infection for a host with genotype $ij$ at site $k$ in the next generation is
\begin{equation}
P_{ij}^k(t+1) = 1 - \exp\left(\mathrm{FOI}_{ij}^k\right).
\end{equation}

Finally, the number of infected individuals in the next generation follows a binomial distribution:
\begin{equation}
\begin{aligned}
S_{ij,I}^k (t+1) &\sim \mathrm{Binom}(S_{ij}^k (t+1), P_{ij}^k),\\
A_{ij,I}^k (t+1) &\sim \mathrm{Binom}(A_{ij}^k (t+1), P_{ij}^k).
\end{aligned}
\end{equation}
Expected number of individuals infected with haplotype $i$ in the next generation is given by a ratio of $\lambda$:
\begin{equation}
\begin{aligned}
S_{ij,i,I}^k(t+1) &= \frac{\lambda_{i, \textrm{total}}^k}{\lambda_{i, \textrm{total}}^k + \lambda_{j, \textrm{total}}^k} S_{ij,I}^k(t+1)\\
A_{ij,i,I}^k(t+1) &= \frac{\lambda_{i, \textrm{total}}^k}{\lambda_{i, \textrm{total}}^k + \lambda_{j, \textrm{total}}^k} A_{ij,I}^k(t+1)
\end{aligned}
\end{equation}



\subsection{Approximate Bayesian Computation}


\section{Results}

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/smc_param.pdf}
\caption{{\bf Need caption.}
Need caption}
\label{fig:smcparam}
\end{figure}

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/simulated_data.pdf}
\caption{{\bf Need caption.}
Need caption}
\label{fig:ivs}
\end{figure}


\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/true_effect.pdf}
\caption{{\bf Need caption.}
Need caption}
\label{fig:teff}
\end{figure}


\section{Discussion}

Here, we offer a slightly different interpretation of the parameter $c_b$. \cite{ashby2015diversity} defines $c_b > 1$ as additional benefits of sex. We can think of it as amount of compensation required to overcome the two fold cost of sex.

\bibliography{redqueen}
\end{document}
