\documentclass{article}\usepackage[]{graphicx}\usepackage[]{color}

\title{Bridging the gap between theory and data: the Red Queen Hypothesis for sex}
\author{Sang Woo Park and Benjamin M Bolker}

\usepackage{tabularx}

\usepackage{amsmath}
\usepackage{natbib}
\usepackage{hyperref}
\bibliographystyle{chicago}
\date{\today}

\usepackage{bm}

\usepackage{afterpage}
\usepackage{pdflscape}

\newcommand{\etal}{\textit{et al.}}

\newcommand{\comment}[3]{\textcolor{#1}{\textbf{[#2: }\textit{#3}\textbf{]}}}
\newcommand{\bmb}[1]{\comment{cyan}{BMB}{#1}}
\newcommand{\swp}[1]{\comment{magenta}{SWP}{#1}}
\newcommand{\citetapos}[1]{\citeauthor{#1}'s \citeyearpar{#1}}

\newcommand{\fref}[1]{Fig.~\ref{fig:#1}}
\begin{document}

\maketitle

\section*{Abstract}

Sexual reproduction persists in nature despite its large cost.
The Red Queen Hypothesis postulates that parasite pressure maintains sexual reproduction in the host population by selecting for the ability to produce rare genotypes that are resistant to infection.
Mathematical models have been used to lay theoretical foundations for the hypothesis; empirical studies have confirmed these predictions.
For example, Lively used a simple host-parasite model to predict that the frequency of sexual hosts should be positively correlated with the prevalence of infection. 
Lively \textit{et al.} later confirmed the prediction through numerous field studies of snail-trematode systems in New Zealand.
In this study, we fit a simple metapopulation host-parasite coevolution model to three data sets, each representing a different snail-trematode system, by matching the observed prevalence of sexual reproduction and trematode infection among hosts.
Using the estimated parameters, we perform a power analysis to test the feasibility of observing the positive correlation predicted by Lively.
We discuss anomalies in the data that are poorly explained by the model and provide practical guidance to both modelers and empiricists.
Overall, our study suggests that a simple Red Queen model can only partially explain the observed relationships between parasite infection and the maintenance of sexual reproduction.

\section{Introduction}

% The evolution of sexual reproduction poses a continuing question \citep{otto2009evolutionary}.
Despite being the dominant mode of reproduction in multicellular organisms \citep{vrijenhoek1998animal}, sexual reproduction entails numerous costs \citep{lehtonen2012many}.
The most commonly mentioned is the cost of producing males \citep{smith1978evolution}:
as males cannot produce offspring, asexual lineages are expected to outgrow their sexual counterparts.
This \emph{two-fold cost of sex} \citep{smith1978evolution} relies on the assumption that everything else is equal.
Then, what drives the observed prevalence of sexual reproduction?

One explanation for the persistence of sexual reproduction is the Red Queen Hypothesis (RQH) \citep{bell1982masterpiece}.
The RQH predicts that sexually reproducing hosts overcome the cost of sex under strong parasitic pressure by producing genetically diverse offspring that can escape infection \citep{jbs1949disease, jaenike1978hypothesis, hamilton1980sex, hamilton1990sexual}.
Limited genetic diversity of asexually reproducing hosts allows for sexual reproduction to be maintained in the host population \citep{ashby2015diversity}.

Much of the theoretical literature has focused on determining qualitative conditions under which parasite selection can maintain sexual reproduction in the host population.
Here, we describe a few important criteria.
First, hosts and parasites must coevolve \citep{bell1982masterpiece}.
Host-parasite coevolution creates a time-lagged selective advantage for rare host genotypes, creating an oscillation in genotypic frequencies and allowing for sexual reproduction to be maintained in the host population \citep{clarke1976ecological,jaenike1978hypothesis, hamilton1980sex, agrawal2001parasites}.
Second, parasites must be highly virulent \citep{may1983epidemiology}.
Although sexual and asexual hosts can coexist at intermediate virulence \citep{howard1994parasitism}, sexual reproduction is unable to provide enough advantage to overcome the cost of sex against avirulent parasites \citep{howard1994parasitism}.
Finally, sexual hosts must be genetically more diverse than asexual hosts, as high clonal diversity may diminish the advantage of sexual reproduction \citep{lively1994selection, lively2010review, ashby2015diversity}.

Some theoretical studies have departed from the classical population genetics framework to study effects of ecological and epidemiological context on the Red Queen dynamics.
A few studies have suggested that incorporating more realistic ecological and epidemiological details may assist in supporting sexual reproduction in the host population \citep{lively2009maintenance, lively2010epidemiological} and maintaining coevolutionary cycles \citep{ashby2014parasitic}.
In contrast, \cite{macpherson2018joint} showed that Red Queen dynamics (i.e., cycles in allele frequencies) fail to persist when explicit epidemiological structure is taken into account with coevolutionary dynamics.
Given the wide range of assumptions underlying these eco-evolutionary models, it may be difficult to understand how different ecological details of these models affect the overall maintenance of sexual reproduction;
however, it is clear is that these details play important roles in shaping coevolutionary dynamics \citep{song2015host, ashby2019understanding, haafke2016eco}.

Many empirical studies have focused on confirming predictions that stem from the RQH.
Typical among them are rapid parasite evolution \citep{rauch2006one}, local adaptation \citep{lively1989adaptation, morran2014experimental, king2011coevolutionary, gibson2016within}, time-lagged selection \citep{buckling2002antagonistic, decaestecker2007host, koskella2009evidence, thrall2012rapid, koskella2013phage, koskella2014bacteria}, and association between parasite prevalence and host reproductive mode \citep{lively1992parthenogenesis, vergara2013geographic, verhoeven2013geographic}. 
A key example is the snail population in New Zealand that serves as an intermediate hosts for trematodes \citep{winterbourn1974larval, mcarthur1976suppression}.
Through decades of work, Lively \etal\ have demonstrated that the population satisfies necessary conditions for the host-parasite coevolutionary dynamics and provides support for the hypothesis (e.g., \cite{lively1987evidence, lively1989adaptation, dybdahl1995host, dybdahl1998host, jokela2009maintenance, vergara2014infection, gibson2016within}).
While field studies provide only indirect evidence for the hypothesis, 
experimental systems can be used to achieve more direct evidence \citep{auld2016sex, slowinski2016coevolutionary}.

Even though the Red Queen Hypothesis has gained both theoretical and empirical support, there still remains a gap between theory and data.
Many theoretical models rely on simplifying assumptions that are not applicable to natural populations and make indirect connections to empirical studies.
For example, none of the Red Queen models reviewed by \cite{ashby2015diversity} make statistical connections to empirical data.
It is unclear how well these models capture coevolutionary dynamics in natural systems.

However, theoretical models can still be used to make \emph{qualitative} predictions about nature.
\cite{lively1992parthenogenesis} initially postulated that infection prevalence should be positively correlated with frequency of sexual hosts and later formalized the idea using a mathematical model \citep{lively2001trematode}.
The prediction has since been confirmed in several empirical studies, most of which are based on the snail-trematode system \citep{lively2002temporal, kumpulainen2004parasites, king2011parasites, vergara2013geographic, mckone2016fine, gibson2016within}.
Surprisingly, such correlation was not observed in a different snail-trematode system \citep{heller1990sexual, ben2005spatial, ben2007temporal, ben2008sex, dagan2013clonal}.

Here, we try to help bridge the gap between theory and data in understanding the role of parasites in maintaining sexual reproduction in host populations and make quantitative inference about empirical systems.
We extend the model used by \cite{lively2010epidemiological} to account for demographic stochasticity and include a simple metapopulation structure.
Then, we fit the model to data sets from three field studies \citep{dagan2013clonal, mckone2016fine, vergara2014infection} using Approximate Bayesian Computation (ABC) to estimate biologically relevant parameters.
We assess model fits and discuss discrepancies between a theoretical model and the observed data.
Using biologically realistic parameters, we test for power (probability of observing a significant effect) to detect a positive correlation between frequency of sexual reproduction and prevalence of infection \cite{lively2001trematode}.
By performing power analysis, we provide guidance for studying the Red Queen Hypothesis and discuss underlying factors that drive the correlation.

\section{Methods}

\subsection{Data}

We analyze three observational data sets \citep{vergara2014infection, mckone2016fine, dagan2013clonal} that represent interactions between freshwater snails and sterilizing trematodes.
The New Zealand snail populations \citep{vergara2014infection, mckone2016fine} consist of \textit{Potamopyrgus antipodarum}, which serves as an intermediate host for trematodes, 
among which \textit{Microphallus} sp. is most commonly found \citep{winterbourn1974larval, lively1987evidence}.
The Israeli snail populations \citep{dagan2013clonal} consist of \textit{Melanoides tuberculata}, which also serves as an intermediate host for trematodes, including \textit{Centrocestus} sp. \citep{ben2005spatial}, \textit{Philophthalmus} sp. \citep{ben2006first}, and \textit{Transversotrema patialense} \citep{ben2005differential}.
Diploid sexual females and polyploid asexual females coexist in both the New Zealand \citep{phillips1989genetics, wallace1992parthenogenesis, dybdahl1995diverse} and the Israeli \citep{samadi1999microsatellite} snail populations.

We choose to analyze snail-trematode systems because they have been extensively studied under the context of the Red Queen Hypothesis.
The snail-trematode systems in New Zealand have been shown to exhibit positive correlation between prevalence of infection and frequency of sexual hosts \citep{lively2002temporal, king2011parasites, vergara2013geographic, mckone2016fine, gibson2016within}; local adaptation \citep{lively1989adaptation, lively2004host, king2011coevolutionary}; negative frequency-dependent selection \citep{dybdahl1995host, dybdahl1998host,jokela2009maintenance, koskella2009evidence}; and periodic selection for sexual reproduction \citep{vergara2014infection, gibson2018periodic}. 
These results are consistent with the host-parasite coevolutionary theory behind the Red Queen Hypothesis.
Therefore, we expect a basic Red Queen model to be able to mimic the observed dynamics reasonably well.

We included the Israeli population because there has been a consistent lack of positive correlation between the prevalence of infection and frequency of sexual hosts \citep{heller1990sexual, ben2005spatial, ben2007temporal, ben2008sex, dagan2013clonal}, which sharply contrasts the observations from the New Zealand population.
Comparison of these two populations with a mathematical model may allow us to better understand the underlying cause of the difference.

The data sets include proportion of infected snails, proportion of sexual/asexual snails, and locations of sampling sites; the data set from \cite{vergara2014infection} also includes sampling year.
Whereas \cite{vergara2014infection} report infection status of snails specific to \textit{Microphallus} sp., \cite{mckone2016fine} and \cite{dagan2013clonal} do not distinguish among different species.
The data sets collected by \cite{dagan2013clonal} and \cite{vergara2014infection} are obtained from their Dryad repositories \citep{dryad_f5t56, dryad_29nk3_2}. 
The data set collected by \cite{mckone2016fine} is extracted from their figure.

\subsection{Model}

We model obligately sexual hosts competing with obligately asexual hosts in a metapopulation by extending the model introduced by \cite{lively2010epidemiological}.
Our model is a discrete time susceptible-infected (SI) model with natural mortality and virulence (defined as reduction in offspring production among infected hosts).
Metapopulation structure is included to model unobserved dynamics among different habitats; equivalently, each population can be considered as a sampling site.
A similar metapopulation model was developed by \cite{lively2018habitat} in order to study local adaptation of parasites.

We do not model the life history of snail-trematode interactions explicitly.
Modeling interactions of multiple trematodes species, each of which go through a different life cycle, with a single snail species is extremely complicated.
Instead, testing a basic model against data will allow us to identify the difference between theory and data more clearly.

All hosts are diploids with two biallelic loci controlling resistance; parasites are assumed to be haploids.
Let $S_{ij}^k(t)$ and $A_{ij}^k(t)$ be the number of sexual and asexual hosts with genotype $ij$ from population $k$ at generation $t$, 
where the subscripts, $i$ and $j$, represents host haplotypes: $i, j \in \{\mathrm{AB}, \mathrm{Ab}, \mathrm{aB}, \mathrm{ab}\}$.
For simplicity, we drop the superscript representing population and write $S_{ij}(t)$ and $A_{ij}(t)$;
every population is governed by the same set of equations unless noted otherwise (e.g., when we account for interaction between populations).
Following \cite{lively2010epidemiological}, the expected genotypic contribution (before recombination or outcrossing) by sexual hosts is given by
\begin{equation}
S_{ij}' = c_b (1-s) \bigg(W_U S_{ij,U} (t) + W_I S_{ij,I} (t)\bigg),
\end{equation}
where $s$ is the proportion of males produced by sexual hosts, and $S_{ij, U}$ and $S_{ij,I}$ are the number of uninfected and infected sexual hosts in a population.
$W_U$ and $W_I$ represent their corresponding fitnesses where virulence is defined as $V = 1-W_I/W_U$.
We allow for the cost of sex to vary by multiplying the growth rate by a scale parameter, $c_b$, where $2/c_b$ corresponds to a two fold cost of sex when $c_b = 1$ \citep{ashby2015diversity}.
Recombination and outcrossing are modeled after incorporating genotypic contributions from other populations.

We define
$$
W_U = \frac{b_U}{1 + a_U N(t)},\,  W_I = \frac{b_I}{1 + a_I N(t)}
$$
where $b_U$ and $b_I$ are the maximum number of offspring produced by uninfected and infected hosts, respectively, and $a_U$ and $a_I$ determine their corresponding strengths of density dependence \citep{smith1973stability, lively2010epidemiological}.
For simplicity, we assume that $a_U = a_I$ so that virulence can be defined strictly in terms of decrease in offspring production: $V = 1- b_I/b_U$.

Asexual hosts are strictly clonal.
The expected genotypic contribution by asexual hosts is given by
\begin{equation}
A_{ij}' = W_U A_{ij,U} (t) + W_I A_{ij,I} (t),
\end{equation}
where $A_{ij, U}$ and $A_{ij,I}$ are the number of uninfected and infected asexual hosts in a population.

A proportion $\epsilon_{\textrm{\tiny mix}}$ of a population mixes with other populations.
The expected number of offspring in the next generation (accounting for contributions from all populations) is given by
\begin{equation}
\begin{aligned}
\mathrm{E}\left(S_{ij}^k(t+1)\right) &= f_{\textrm{\tiny sex}}\left((1 - \epsilon_{\textrm{\tiny mix}}) \left(S_{ij}^k\right)' + \frac{\epsilon_{\textrm{\tiny mix}}}{n_{\textrm{\tiny pop}}-1} \sum_{h \neq k} \left(S_{ij}^h\right)'\right),\\
\mathrm{E}\left(A_{ij}^k(t+1)\right) &= (1 - \epsilon_{\textrm{\tiny mix}}) \left(A_{ij}^k\right)' + \frac{\epsilon_{\textrm{\tiny mix}}}{n_{\textrm{\tiny pop}}-1} \sum_{h \neq k} \left(A_{ij}^h\right)',\\
\end{aligned}
\end{equation}
where $f_{\textrm{\tiny sex}}(x)$ is the function that models sexual reproduction, including recombination probability $r_{\textrm{\tiny host}}$ and outcrossing, and $n_{\textrm{\tiny pop}}$ is the number of populations modeled.
The total number of sexual and asexual hosts in the next generation is given by a Poisson random variables with means specified previously. We also allow for stochastic migration in order to avoid fixation:
\begin{equation}
\begin{aligned}
S_{ij}^k(t+1) &\sim \mathrm{Poisson}\left(\lambda=\mathrm{E}\left(S_{ij}^k(t+1)\right)\right) + \mathrm{Bernoulli}\left(p=p_{ij,\textrm{\tiny sex}}\right),\\
A_{ij}^k(t+1) &\sim \mathrm{Poisson}\left(\lambda=\mathrm{E}\left(A_{ij}^k(t+1)\right)\right) + \mathrm{Bernoulli}\left(p=p_{ij, \textrm{\tiny asex}}\right),
\end{aligned}
\end{equation}
where $p_{ij,\textrm{\tiny sex}}$ and $p_{ij, \textrm{\tiny asex}}$ are the probabilities of a single sexual or asexual host with genotype $ij$ entering a subpopulation.

Infection is modeled using the matching alleles model \citep{otto1998evolution}.
We assume that snails are equally susceptible to parasites that match either haplotype.
However, parasites must match the host haplotype at both loci in order to successfully infect a host.
The expected number of infected hosts that are infected with parasite with genotype $i$ at generation $t$ is given by:
\begin{equation}
I_{i}(t) = f_{\textrm{\tiny mutation}} \left( \sum_{j}  \bigg( S_{ij,i,I}(t) + A_{ij,i,I}(t)\bigg)\right).
\end{equation}
$S_{ij,i,I}(t)$ and $A_{ij,i,I}(t)$ represent the expected numbers of sexual and asexual hosts that have genotype $ij$ and are infected with parasites that have genotype $i$.
The function $f_{\textrm{\tiny mutation}}$ models mutation at a single locus with probability $r_{\textrm{\tiny parasite}}$ \citep{ashby2015diversity}. 
We further account for stochastic migration with probability $p_{i, \textrm{\tiny parasite}}$ to avoid fixation:
\begin{equation}
I_{i}'(t) \sim I_i(t) + \mathrm{Bernoulli}\left(p=p_{ij, \textrm{\tiny parasite}}\right).
\end{equation}


The total expected number of infectious contacts made by infected hosts within a population is given by $\lambda_i^k = \beta^k {I_i'}^k(t)$, where $\beta^k$ is the transmission rate of each population. 
We model mixing between subpopulations by allowing infected hosts to make contact with susceptible hosts in other populations;
we assume that a proportion $\epsilon_{\textrm{\tiny mix}}$ of the infectious contacts is equally distributed among other subpopulations.
The total amount of infectious contact, coming from hosts that carry genotype $i$ parasite, that is received by susceptible hosts in population $k$ is given by
\begin{equation}
\lambda_{i, \textrm{\tiny total}}^k = (1 - \epsilon_{\textrm{\tiny mix}}) \lambda_i^k + \frac{\epsilon_{\textrm{\tiny mix}}}{n_\textrm{\tiny pop}-1} \sum_{l \neq k} \lambda_i^l
\end{equation}
Then, the force of infection that a susceptible host with genotype $ij$ experiences in generation $t+1$ is given by
\begin{equation}
\mathrm{FOI}_{ij}^k = \frac{\lambda_{i, \textrm{\tiny total}}^k  + \lambda_{j, \textrm{\tiny total}}^k}{2 N^k(t+1)},
\end{equation}
where $N^k(t+1) = \sum_{i,j} S_{ij}^k(t+1) + A_{ij}^k(t+1)$ is the total number of hosts in generation $t+1$.
The probability that a susceptible host with genotype $ij$ in population $k$ becomes infected in the next generation is given by
\begin{equation}
P_{ij}^k(t+1) = 1 - \exp\left(\mathrm{FOI}_{ij}^k\right).
\end{equation}

Finally, the number of infected hosts in the next generation is determined by a binomial process:
\begin{equation}
\begin{aligned}
S_{ij,I}^k (t+1) &\sim \mathrm{Binom}\left(S_{ij}^k (t+1), P_{ij}^k(t+1)\right),\\
A_{ij,I}^k (t+1) &\sim \mathrm{Binom}\left(A_{ij}^k (t+1), P_{ij}^k(t+1)\right).
\end{aligned}
\end{equation}
The expected number of infected hosts that have genotype $ij$ and are infected by parasites with genotype $i$ in the next generation is proportional to the amount of infectious contact that was made in the current generation:
\begin{equation}
\begin{aligned}
S_{ij,i,I}^k(t+1) &=  \frac{2^{\delta_{ij}} \lambda_{i, \textrm{\tiny total}}^k}{\lambda_{i, \textrm{\tiny total}}^k + \lambda_{j, \textrm{\tiny total}}^k} S_{ij,I}^k(t+1)\\
A_{ij,i,I}^k(t+1) &=  \frac{2^{\delta_{ij}} \lambda_{i, \textrm{\tiny total}}^k}{\lambda_{i, \textrm{\tiny total}}^k + \lambda_{j, \textrm{\tiny total}}^k} A_{ij,I}^k(t+1)
\end{aligned}
\end{equation}
where a $\delta_{ij}$ is a Kronecker delta ($\delta_{ij}$ equals 1 when $i = j$ and 0 otherwise).

\subsection{Simulation design and parameterization}

Many Red Queen models have focused on competition between a single asexual genotype and multiple sexual genotypes or have assumed equal genetic diversity between asexual and sexual hosts (see \cite{ashby2015diversity} for a review of previous Red Queen models) but neither of these assumptions is realistic.
In contrast, \cite{ashby2015diversity} adopted a more realistic approach by incorporating stochastic external migration of an asexual genotype to a population and allowing for asexual genetic diversity to vary over time.
Here, we combine these methods.
We allow for stochastic external migration of asexual hosts with different genotypes into the system but fix the number of asexual genotypes (denoted by $G_{\textrm{\tiny asex}}$) that can be present in the system. 
Since sexual hosts have limited genetic diversity in our model (diploid hosts with two biallelic loci yields a total of 10 genotypes), allowing for unlimited migration of asexual hosts will cause the sexual population to be easily outcompeted by the asexual population.
Limiting asexual genetic diversity allows us to account for the intrinsic difference in sexual and asexual diversity and make a compromise between simple and realistic models.
At the beginning of each simulation, a pool of $G_{\textrm{\tiny asex}}$ asexual genotypes are sampled at random from the entire genotypic space; these are the genotypes that are available for external immigration into subpopulations.
Then, we use our model-based comparison with data to estimate $G_{\textrm{\tiny asex}}$ to test whether the data are consistent with the dynamics of a model that allows for more than a single asexual clone.

To account for differing number of sexual and asexual genotypes, we let 
\begin{equation}
\begin{aligned}
p_{ij, \textrm{\tiny sex}} &= 1-(1-p_{\textrm{\tiny host}})^{1/G_\textrm{\tiny sex}},\\
p_{ij, \textrm{\tiny asex}} &=
\begin{cases}
1-(1-p_{\textrm{\tiny host}})^{1/G_\textrm{\tiny asex}} & \text{if } ij \in \{\text{asexual genotypes}\} \\
0 & \text{otherwise}
\end{cases},
\end{aligned}
\end{equation}
where $p_{\textrm{\tiny host}}$ is the probability that at least one sexual or asexual host enters the population in a generation. We scale the probability of an infected host carrying parasite genotype $i$ in a similar way for interpretability:
\begin{equation}
p_{i, \textrm{\tiny parasite}} = 1 - (1-p_{\textrm{\tiny infected}})^{1/G_\textrm{\tiny parasite}},
\end{equation}
where $p_{\textrm{\tiny infected}}$ is the probability that at least one infected host enters the population in a generation.
The number of parasite genotypes $G_\textrm{\tiny parasite}$ is equal to 4 (because parasites are assumed to be haploids with two loci).

Each simulation consists of 40 subpopulations. Every subpopulation is initialized with 2000 sexual hosts, of which 80 are infected. 
They are assumed to be in Hardy-Weinberg equilibrium where allele frequencies in each locus is equal to 0.5. 
The transmission rate, $\beta^k$, is randomly drawn for each subpopulation from a Gamma distribution with mean $\beta_{\textrm{\tiny mean}}$ and coefficient of variation $\beta_{\textrm{\tiny CV}}$ in order to account for variation in infection rates among subpopulations. 
The simulation runs for 500 generations without introduction of asexuals. At generation 501, 10 asexual hosts of a single genotype are introduced to each population (the asexual genotype introduced can vary across population) and simulation runs for a further 600 generations while allowing for stochastic migration of asexuals.

\subsection{Approximate Bayesian Computation}

We use Approximate Bayesian Computation (ABC) to estimate parameters from data \citep{toni2009approximate}.
ABC relies on comparing summary statistics of observed data with those of simulated data; it is particularly useful when the exact likelihood function is not available.
We consider the mean proportion of infected and sexually reproducing snails in the system and variation in these proportions --- measured by coefficient of variation (CV) --- across space (population) and time as our focal summary statistics.
As \cite{dagan2013clonal} and \cite{mckone2016fine} only reported the proportion of males, the proportion of sexual hosts is assumed to be twice the proportion of males.

CV across space is calculated by first calculating mean proportions by averaging across time (generation) for each site (subpopulation) and then taking the the CV of these mean proportions.
CV across time (generation) is calculated by first averaging proportions across space (subpopulation) at each generation and then taking the CV.
For purely spatial data (\cite{dagan2013clonal} and \cite{mckone2016fine}), CV across space is calculated without averaging across time.

Because ABC is a Bayesian method, we must specify prior distributions for all parameters.
We use weakly informative priors for all estimated parameters except $c_b$, a scale parameter for the cost of sex (see Table~\ref{tb:param} for prior distributions used and parameters assumed).
The prior distribution for the scale parameter is chosen so that 95\% prior quantile range of cost of sex (2/$c_b$) is approximately equal to the 95\% confidence interval reported by \cite{gibson2017two}.
For brevity, all other parameters are assumed to be fixed.
While it is a common practice to fix several parameters of a Red Queen model (e.g., \cite{lively2010epidemiological, ashby2015diversity, haafke2016eco, ashby2019understanding}), failing to explicitly account for uncertainty in all parameters of a model can lead to overly confident conclusions \citep{elderd2006uncertainty}.

\afterpage{
\clearpage
\begin{landscape}
\begin{table}[h]
\centering
\begin{tabular}{c|p{5cm}|c|c}
\hline
\textbf{Notation} & \textbf{Description} & \textbf{Prior distribution/parameter values} & \textbf{Source}\\
\hline
$\beta_{\textrm{\tiny mean}}$ & Mean transmission rate & $\mathrm{Gamma}(k=2, \theta=10)$ & Assumption\\
$\beta_{\textrm{\tiny CV}}$ & CV transmission rate & $\mathrm{Gamma}(k=2, \theta=0.5)$ & Assumption\\
$V$ & Virulence & $\mathrm{Beta}(\alpha=6, \beta=2)$ & Assumption\\
$\epsilon_{\textrm{\tiny mix}}$ & Mixing proportion & $\mathrm{Beta}(\alpha=1, \beta=9)$ & Assumption\\
$G_{\textrm{\tiny asex}}$ & Number of asexual genotypes & $1 + \mathrm{BetaBinomial}(N=9, p=3/9, \theta=5)$ & Assumption\\
$c_b$ & Cost of sex scale & $\mathrm{LogNormal}(\mu=-0.07, \sigma=0.09)$ & \cite{gibson2017two}\\
$s$ & Proportion of male offspring produced & 0.5 & Assumption\\
$b_U$ & Number of offspring produced by an uninfected host & 20 & \cite{lively2010epidemiological}\\
$b_I$ & Number of offspring produced by an infected host & $(1-V) b_U$ & \cite{lively2010epidemiological}\\
$a_U$ & Density dependent effect coefficient of uninfected hosts & 0.001 & \cite{lively2010epidemiological}\\
$a_U$ & Density dependent effect coefficient of infected hosts & 0.001 & \cite{lively2010epidemiological}\\
$r_{\textrm{\tiny host}}$ & Host recombination probability & 0.2 & \cite{lively2010epidemiological}\\
$r_{\textrm{\tiny parasite}}$ & Parasite mutation probability & 0.05 & Assumption\\
$p_{\textrm{\tiny host}}$ & Probability that at least one sexual and asexual host enters the population & 0.1 & Assumption\\
$p_{\textrm{\tiny infected}}$ & Probability that at least one infected host enters the population & 0.02 & Assumption\\
\hline
\end{tabular}
\caption{
\textbf{Parameter descriptions and values}.
Parameters with prior distributions are estimated via Approximate Bayesian Computation (ABC).
Parameters $k$ and $\theta$ in Gamma distribution represent shape and scale parameters where mean and squared CV are given by $k \theta$ and $1/k$, respectively.
$\alpha$ and $\beta$ in Beta distribution represent shape parameters where mean and squared CV are given by $\alpha/(\alpha+\beta)$ and $\beta/(\alpha^2 + \alpha \beta + \alpha)$.
$N$, $p$ and $\theta$ in Beta binomial distributions represent number of trials, probability of success, and overdispersion parameters \citep{morris1983natural}.
Parameters $\mu$ and $\sigma$ in a log-normal distribution represent mean and standard deviation on a log scale.
All other parameters are fixed throughout simulations.
}
\label{tb:param}
\end{table}
\end{landscape}
}

We use the Population Monte Carlo approach \citep{turner2012tutorial}, which allows for efficient sampling while ensuring that final result still converges to a correct (approximate) Bayesian posterior.
The Population Monte Carlo approach begins with a basic ABC.
For each random parameter sample drawn from the prior distribution, the model is simulated and a sample of subpopulation is drawn from the simulated system, equal to the number of sites collected in a study.
Summary statistics are calculated based on the last 100 generations out of 1100 generations; current parameter set is accepted if the distance between simulated and observed data is less than a specified tolerance value.
Distance is measured by the sum of absolute differences in summary statistics between simulated and observed data. This process is repeated until 100 parameter sets are accepted.

After the first run ($t=1$), equal weights ($w_{i,1}=1/100$) are assigned to each accepted parameter set $\bm\theta_{i, 1}$ ($i = 1, 2, \dots, 100$).
A weighted random sample ($\bm\theta^\ast$) is drawn from the accepted parameters of the previous run ($t-1$) with weights $w_{i,t-1}$.
For any run $t > 1$, a parameter sample ($\bm\theta_{i, t}$) is proposed from a multivariate normal distribution with a mean $\bm\theta^\ast$ and a variance covariance matrix that is equal to $\sigma_{t-1}^2=2 \mathrm{Var}(\bm\theta_{1:N, t-1})$, where $\mathrm{Var}(\bm\theta_{1:N, t-1})$ is the weighted variance covariance matrix of the accepted parameters from the previous run.
$N$ is the total number of accepted parameters from the previous run.

The parameter $G_{\textrm{\tiny asex}}$ is rounded to the nearest integer and the model is simulated.
If a proposed parameter is accepted, the following weight is assigned:
$$
w_{i,t} = \frac{\pi(\bm\theta_{i, t})}{\sum_{i=1}^{100} w_{j, t-1} q(\bm\theta_{j, t-1} | \bm\theta_{i,t}, \sigma_{t-1}^2)}
$$
where $\pi(\cdot)$ is a prior density and $q(\cdot | \bm\theta_{i,t}, \sigma_{t-1}^2)$ is a multivariate normal density with mean $\bm\theta_{i,t}$ and variance covariance matrix $\sigma_{t-1}^2$.
For each run, 100 parameters are accepted and weights are normalized at the end to sum to 1.

For each observed data set, we perform 4 runs with decreasing tolerance.
First three tolerance values are chosen in a decreasing sequence to reach the final step quicker.
The tolerance value of the final run is chosen so that a parameter set will be accepted if its each simulated summary statistic deviates from the corresponding observed summary statistic by 0.1 units on average.
For spatial data \citep{dagan2013clonal, mckone2016fine}, four summary statistics are compared: mean proportion of infected and sexually reproducing hosts and CV in these proportions across populations.
Tolerance values of 1.6, 0.8, 0.6 and 0.4 are used for each run.
For spatiotemporal data \citep{vergara2014infection}, we additionally compare the CV of proportions of infected and sexually reproducing hosts across generations.
In this case, larger tolerance values (2.4, 1.2, 0.9 and 0.6) are used for each run to account for higher number of summary statistics being compared.

All statistical results are weighted by parameter weights of the final run; 
confidence intervals are obtained by taking weighted quantiles. 
However, it is always not possible to obtain the exact 95\% confidence interval when the smallest (or the largest) value has a weight greater than 0.025. 
In these cases, we take the lowest (or largest) possible quantile that is closest to the 2.5\% (or the 97.5\%) quantile.
While this procedure results in a slightly narrower confidence interval, the rounding errors in quantiles are negligible. 
This rule leads to one instance where we use the 2.82\% quantile instead of the 2.5\% quantile as the lower end of a confidence interval.

\subsection{Power analysis}

Using estimated parameters for each data set, we calculate the power to detect a significant positive correlation between infection prevalence and frequency of sexual hosts.
For each parameter sample from the final run of the ABC, 10 simulations are run.
For each simulation, we take the last two generations --- assuming that a year contains two snail generations (estimated generation time is 4-9 months \citep{neiman2005variation}) --- from the simulation and choose $n$ populations at random from 40 simulated populations.
For each selected population, hosts are divided into four categories based on their infection status (infected/uninfected) and reproductive mode (asexual/sexual),
and mean proportion of hosts in each category is calculated by averaging over two generations.
Independent multinomial samples of size $m$ are drawn from each selected population based on the proportions  in each four categories. 
Correlation between proportion of infected hosts and proportion of sexual hosts is tested using the Spearman's rank correlation at a 5\% significance level.

While we do not explicitly try to match observed correlations, this procedure is essentially a retrospective power analysis.
The main purpose of a power analysis is to design an experiment and determine an appropriate sample size \citep{cohen1992statistical}.
If instead, power is calculated based on an observed effect size after an experiment, the estimated power is likely to be correlated with the observed significance of the result;
therefore, interpreting significance (or a lack of significance) of a statistical result based on a retrospective power analysis is inappropriate \citep{goodman1994use, senn2002power}.

Here, we do not try to justify the significant correlation observed by \citep{mckone2016fine} or the nonsignificant correlation observed by \citep{dagan2013clonal} based on this power analysis. 
Instead, we perform a power analysis to better understand how a study design or the dynamics of the model might affect the correlation between the prevalence of infection and sexual reproduction while using biologically realistic parameters.

\section{Results}

First, we compare observed summary statistics with either fitted or predicted summary statistics (\fref{smcsumm}).
\emph{Fitted} (filtered) summary statistics (\fref{smcsumm}) are values that have been accepted by the ABC procedure as being sufficiently close to observed summary statistics;
these can be interpreted as model-based estimates of the summary statistics (and associated confidence intervals) of the study sites.
\emph{Predicted} (unfiltered) summary statistics, in contrast (\fref{ivs}), use parameters from the posterior distribution sampled by ABC, but generate new simulations of the dynamics and calculate summary statistics from randomly selected subpopulations.
These values represent summary statistics sampled from the full distribution of coevolutionary dynamics that are consistent with the observed dynamics; they represent the expected range of dynamics from randomly sampled host-parasite metapopulations with a given distribution of parameters.
Because they include an additional level of dynamical variability, the predicted summary statistics range much more widely than the fitted summary statistics.

Our simple meta-population Red Queen model can capture observed variation in infection prevalence and frequency of sexual hosts reasonably well;
both temporal and spatial variation (measured by CV across mean proportions) are well-matched by the model. 
However, as model fitting is performed by minimizing the sum of absolute distance between observed and simulated summary statistics, our method does not guarantee that all summary statistics are equally well-fitted.
The model tends to overestimate mean proportion of infected hosts.
The observed mean proportions of infected hosts are 0.175 \citep{dagan2013clonal}, 0.051 \citep{mckone2016fine}, and 0.440 \citep{vergara2014infection}, 
whereas their ABC-based estimates are 0.240 (95\% CI: 0.174 - 0.286), 0.313 (95\% CI: 0.233 - 0.404), and 0.542 (95\% CI: 0.360 - 0.730), respectively.
The model underestimates the mean proportion of sexual hosts for \cite{dagan2013clonal} and \cite{vergara2014infection}.
Observed mean proportions of sexual hosts are 0.045 and 0.704, respectively, whereas their ABC-based estimates are 0.026 (95\% CI: 0.007 - 0.048) and 0.596 (95\% CI: 0.441 - 0.679).

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/smc_summary.pdf}
\caption{{\bf Summary statistics of the observed data vs. distribution of summary statistics of the simulated data from the posterior samples.}
Dotted horizontal lines represent observed summary statistics.
Violin plots show weighted distribution of fitted summary statistics (i.e., summary statistics that were accepted during ABC). 
Error bars show 95\% weighted quantiles of predicted summary statistics.
The weights correspond to the posterior distribution weights from ABC.
For each posterior sample, 10 simulations are run and each simulated system is sampled at random 100 times so that each sample consists of the same number of populations as in the fitted data.
All univariate summary statistics are matched reasonably well, except for the mean proportion of infected hosts in \cite{mckone2016fine} (indicated by an asterisk). 
}
\label{fig:smcsumm}
\end{figure}

To further diagnose the fit, we compare the predicted relationships between the mean proportion of infected hosts and the mean proportion of sexual hosts across subpopulations with the observed data (\fref{ivs}).
% we consider infection prevalence and frequency of sexual hosts averaged over the last two generations to remain consistent with power analyses.
Despite its accuracy in reproducing summary statistics reported by \cite{dagan2013clonal}, 
our model poorly captures the relationship between the mean proportion of infected hosts and the mean proportion of sexual hosts (\fref{ivs}; \cite{dagan2013clonal}).
The model predicts sexual reproduction to be well maintained when infection prevalence is high ($> 40\%$) whereas
the observed data \citep{dagan2013clonal} suggests that sexual reproduction can only be supported when infection prevalence is low ($< 20\%$).
\cite{mckone2016fine} also found sexually reproducing snails in sites with low infection prevalence ($< 20\%$).

On the other hand, \citetapos{vergara2014infection} data set suggests that sexually reproducing populations are likely to have a relatively high prevalence of infection (approximately 20\% - 80\%).
This observation is broadly consistent with our model prediction. 
Most of the data points fall within the range of model predictions (\fref{ivs}; \cite{vergara2014infection}).
However, there is one site in which more than 90\% of the snails were found to be sexual throughout the study period of 5 years \citep{vergara2014infection};

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/simulated_data2.pdf}
\caption{{\bf Predicted relationship between mean infection prevalence and mean proportion of sexual hosts in each population.}
For each posterior sample, 10 simulations are run.
For each population within a simulation, mean infection prevalence and mean proportion of sexual hosts is calculated by averaging across last two generations. 
Each point, consisting of mean infection prevalence and meal proportion of sexual hosts, is assigned the weight of the parameter used to simulated the population.
The density at each cell is calculated as the sum of weights of the points that lie within it.
Cell densities are normalized by dividing by the maximum grid densities for each fit.
Dashed contours are calculated at $\log(\textrm{density}) = -4$.
Open triangles represent observed data; proportion of sexual hosts is computed by doubling proportion of male hosts.
}
\label{fig:ivs}
\end{figure}

There is a high posterior density region in which the proportion of infected hosts remains almost constant (around 0.5) among subpopulations while the proportion of sexual hosts can range from 0 to 0.3 (visible in the fits to \cite{mckone2016fine}).
As transmission rate ($\beta$) increases, selection for sexual hosts increases but the increasing number of resistant offspring prevents further infection from occurring and can decrease overall infection prevalence.
This trend is consistent with previous results of \cite{lively2001trematode} who noted that there is a region in which either sexual and asexual reproduction can be selected exclusively under the same infection prevalence.

The proportion of sexual hosts decreases when infection prevalence is extremely high (visible in the fits to \cite{vergara2014infection}).
This pattern can be explained by the decrease in fitness of sexual hosts with increase in infection prevalence, predicted by \cite{ashby2015diversity}. 
The same pattern can be seen in earlier work by \cite{lively2010epidemiological} although it was not discussed.

\fref{smcparam} presents parameter estimates.
Keeping in mind that we do not obtain good fits to data from \cite{dagan2013clonal} and \cite{mckone2016fine}, we still find that high virulence and a low ratio of asexual to sexual genetic diversity are necessary to explain the observed dynamics.
Moreover, we are able to capture observed differences in mean and variation in infection prevalence among studies in our estimates of transmission rate parameters ($\beta_{\textrm{\tiny mean}}$ and $\beta_{\textrm{\tiny CV}}$).

Our fits to \cite{mckone2016fine} suggest that the scale parameter for the cost of sex, $c_b$, should be higher than our prior assumption based on \cite{gibson2017two} that estimated the cost of sex to be 2.14 (95\% CI: 1.81 - 2.55).
\cite{ashby2015diversity} defined $c_b$ as additional costs and benefits of sex, where $c_b=1$ corresponds the two fold cost.
Under their interpretation, our estimate of $c_b$ corresponds to a slightly lower estimate of cost of sex: 1.95 (95\% CI: 1.68 - 2.4). 
(We propose an alternate interpretation to this parameter estimate below).

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/posterior.pdf}
\caption{{\bf Parameter estimates from Sequential Monte Carlo Approximate Bayesian Computation.}
We estimate mean transmission rate ($\beta_{\textrm{\tiny mean}}$), CV in transmission rate ($\beta_{\textrm{\tiny CV}}$), virulence ($V$), mixing proportion ($\epsilon_{\textrm{\tiny mean}}$), number of asexual genotypes ($G_{\textrm{\tiny asex}}$), and a scale parameter for cost of sex ($c_b$).
Violin plots represent weighted distribution of 100 posterior samples obtained from ABC.
Violin plots for prior distribution is obtained by drawing 10000 random parameter samples from the prior distribution.
$G_{\tiny \textrm{asex}}$ is a discrete variable but is drawn on a continuous scale for convenience.
See Table~\ref{tb:param} for full parameter descriptions and their prior distributions.
}
\label{fig:smcparam}
\end{figure}

Finally, our power analyses suggest that there is high power to detect a positive correlation between infection prevalence and frequency of sexual hosts in the systems studied by \cite{dagan2013clonal} and \cite{mckone2016fine} (\fref{power}).
Such high power predicted for \cite{dagan2013clonal} is particularly surprising given that they were not able to observe the expected correlation.
This discrepancy implies that the snail populations studied by \cite{dagan2013clonal} show sufficient variation in infection prevalence in order for the correlation to be observed under pure Red queen selection, but other underlying factors that are neglected by our model may have caused the populations to deviate from their expected behaviours.
On the other hand, our model predicts low power for detecting the positive correlation for the system studied by \cite{vergara2014infection} (\fref{power}).

Overall, our analysis suggests that increasing the number of study sites is a more effective way to increase power than increasing the number of samples per site (\fref{power}).
The correlation between the proportion of infected and sexual hosts depends on the spatial distribution of parasite infection; 
increasing the number of sites can better capture this distribution and increase the power to observe the correlation.
On the other hand, collecting more samples from a site only gives us a more accurate estimate of the two proportions for that specific site; it does not tell us how these proportions are distributed across space.
As a result, the power quickly saturates once we have a sufficient number of samples ($> 100$) to estimate the two proportions reliably at each site.

While we originally planned to perform power analysis using Spearman's rank correlation,
we repeated the analysis using Pearson's correlation after applying arcsine square root transformation \citep{lively1992parthenogenesis} to see whether this procedure improves power.
Using Pearson correlation gives slightly higher power to detect the positive correlation between frequency of sexual hosts and infection prevalence (see Appendix).

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/power.pdf}
\caption{{\bf Power to detect a statistically significant positive correlation between infection prevalence and frequency of sexual hosts.}
Spearman's rank correlation was used to test for correlation between infection prevalence and frequency of sexual hosts in simulated data from the posterior distributions.
}
\label{fig:power}
\end{figure}

\section{Discussion}

A simple metapopulation model for host-parasite coevolution is able to match the observed prevalence of sexual reproduction and trematode infection in snail populations (\fref{smcsumm}).
A direct comparison between a model and a data set allows us to infer biologically meaningful parameters of a model (\fref{smcparam}) and estimate the power to observe a significant, positive correlation between the proportion of infected hosts and the proportion of sexual hosts (\fref{power}), as predicted by \cite{lively1992parthenogenesis}.
However, discrepancies between the model predictions and observed data suggest that a simple host-parasite coevolution model may not be able to sufficiently explain maintenance of sexual reproduction observed in snail populations (\fref{ivs}).

A model that fits poorly can sometimes tell us more about a biological system than a model that fits well.
Our model clearly failed to match the data presented by \cite{dagan2013clonal} (\fref{ivs}).
The snail populations studied by \cite{dagan2013clonal} live in qualitatively different environments from the two other snail populations that we considered \citep{vergara2014infection, mckone2016fine}.
For example, their habitats are subject to seasonal flash floods \citep{ben2007temporal}, which can affect reproductive strategies of snails and interfere with the host-parasite coevolution \citep{lytle2000biotic, ben2007temporal}.
As a result, a positive correlation between infection prevalence and frequency of sexual reproduction could not be detected from the system even though our model (which does not incorporate disturbance) predicts high power.
These results suggest that a significant correlation (or the lack of it) between infection prevalence and the frequency of sexual reproduction may not provide sufficient evidence for (or against) the role of parasites in maintaining sexual reproduction in the host population.

The model fits to \cite{dagan2013clonal} and \cite{mckone2016fine} suggest that the cost of sex can be overcome and sexual reproduction can be maintained only if infection prevalence is much higher than the observed prevalence (\fref{ivs}).
In other words, the benefit of producing offspring with novel genotype is relatively small when infection prevalence is low.
In order to support sexual reproduction at lower infection prevalence, the benefit of sex must be greater or other mechanisms must compensate for the difference.
As our model relies on a simple structure and strong parametric assumptions, an additional benefit of sex can only be provided by lowering the cost of sex (i.e., increasing the scale parameter, $c_b$).

The simple structure of the model and limited genetic diversity can explain the discrepancy between model prediction and the observed data by \cite{mckone2016fine}.
Here, we assumed that host resistance to infection is determined entirely by two biallelic loci, resulting in 10 possible genotypes.
However, it is unlikely that such simple model can capture the genetic interaction between hosts and parasites observed in nature.
Although exact genetic architecture that determines trematode infection in snails (e.g., the number and inheritance of loci involved in parasite resistance) is not known, the documented genetic diversity of snails is far greater than what our model assumes \citep{fox1996genetic, king2011parasites, dagan2013clonal}.
Increasing the maximum possible genetic diversity of the model would have allowed sexual hosts to escape infection more easily and maintained sexual reproduction at a lower prevalence of infection \citep{lively2010effect, king2012does, ashby2015diversity}.

While the positive correlation between frequency of sexual hosts and prevalence of infection can provide evidence for the effect of the parasite on the maintenance of sexual reproduction in the host population, it does not fully capture the coevolutionary process.
In particular, the positive correlation represents a contrast between populations that undergo Red Queen dynamics and those that do not (rather than a continuous relationship between the frequency of sexual hosts and prevalence of infection):
low-risk populations will be dominated by asexual individuals whereas high-risk populations will consist of both sexual and asexual individuals.
\cite{lively2001trematode} gave a similar explanation and predicted that large variation in the risk of infection is required to observe the predicted correlation.

The similarity between the correlations predicted by the model and the correlations observed in nature may provide us more confidence that the observed populations provide evidence for the Red Queen Hypothesis (\fref{effect}).
However, the strength of correlation is not a direct measure of the strength of selection imposed on the host population.
Moreover, when populations are actively coevolving, wide range of correlations can be detected (\fref{effect}: \cite{vergara2014infection}).
While simple statistical summaries, such as a correlation coefficient, are easily accessible,
more sophisticated and mechanistic statistical models that yield biologically interpretable parameters may be better suited for understanding the effect of parasites on the maintenance of sexual reproduction in the host population.

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/effect_size.pdf}
\caption{{\bf Predicted vs. observed correlation between infection prevalence and frequency of sexual hosts.}
Violin plots show weighted distribution of predicted strength of correlation based on ABC fits.
Predicted correlation was measured for each simulation from the posterior by taking into account the last two generations from each simulated population.
Triangles and circles represent observed correlation sizes from previous studies.
\cite{dagan2013clonal} and \cite{vergara2014infection} do not report the size of the correlation; correlations are re-calculated from their data sets using both Pearson (after applying arcsine square root transformation) and Spearman correlations.
}
\label{fig:effect}
\end{figure}

Mathematical models have used extensively to build theoretical foundations for the evolution of sex but only a few models have been confronted with data.
The idea of the two fold cost of sex was only recently tested directly by fitting a theoretical model to observed data \citep{gibson2017two}.
Making statistical inference on the observed systems and testing theoretical models against data may provide deeper insight into underpinnings of maintenance of sexual reproduction.

\section{Acknowledgements}

We thank SHARCnet for providing computational resources. 

\section{Funding}

This work is supported by The Natural Sciences and Engineering Research Council, Undergraduate Student Research Award (to SWP).

\bibliography{redqueen}

\pagebreak
\section*{Appendix}

\renewcommand\thefigure{A\arabic{figure}}    
\setcounter{figure}{0}   

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/power_lm.pdf}
\caption{{\bf Power to detect a statistically significant positive correlation between infection prevalence and frequency of sexual hosts.}
Pearson correlation was used to test for correlation between square root arcsine transformed infection prevalence and frequency of sexual hosts in simulated data from the posterior distributions.
}
\label{fig:power_lm}
\end{figure}

\end{document}
