\documentclass{article}
\title{Quantifying the effects of parasites on the maintenance of sex}
\author{Sang Woo Park, Benjamin M Bolker}

\usepackage{tabularx}

\usepackage{amsmath}
\usepackage{natbib}
\usepackage{hyperref}
\bibliographystyle{chicago}
\date{\today}

\newcommand{\etal}{\textit{et al.}}

\newcommand{\fref}[1]{Fig.~\ref{fig:#1}}

\begin{document}

\maketitle

\section{Introduction}

Sexual reproduction is followed by numerous costs \citep{lehtonen2012many},
and these costs are often attributed to cost of producing males \citep{smith1978evolution}.
As males cannot produce offspring, a sexually reproducing population should be outgrown by an asexually reproducing population growing twice as fast (given that the sexual population produces 50\% males and 50\% females) --
hence the infamous \emph{two-fold cost of sex} \citep{smith1978evolution}.
However, the two-fold cost of sex relies on the assumption that all else is equal, and this is not necessarily true in nature.
In fact, sexual reproduction is much more common in nature \citep{vrijenhoek1998animal}.
So what drives sex to persist?

One explanation for sex is the Red Queen Hypothesis \citep{bell1982masterpiece}.
The Red Queen Hypothesis predicts sexually reproducing organisms to overcome the two-fold cost of sex under strong parasite selection by producing genetically diverse offspring that are resistant to infection \citep{jbs1949disease, jaenike1978hypothesis, hamilton1980sex}.
Hosts with rare genotypes are expected to result in higher fitness and become common eventually.
Furthermore, parasites coevolves with the host population to infect the common "rare" genotype [CITE],
and negative frequeny dependnece drives an oscillation in genotypic frequencies in host and parasite populations \citep{clarke1976ecological, hamilton1980sex}.

Since the Red Queen Hypothesis has been postulated, much of theoretical work has focused on figuring out conditions under which parasites can maintain sexual reproduction.

On the other hand, empiricists have tried to quantify the Red Queen.

While the Red Queen Hypothesis is well supported both empirically and theoretically, there still remains a gap between theory and data.
Many models for Red Queen Hypothesis rely on simplifying assumption that are not applicable to natural populations \citep{ashby2015diversity}.
For example, 

Recent studies have shown that ecological feedbacks can allow sex to be maintained more easily [CITE].
We find that a simple model can capture the dynamics 
Here, we try to bridge this gap further.

% \cite{lively2001trematode} successfully applied ???. Using a mathematical model, he predicted that prevalence of infection should be positively correlated with prevalence of sex given large variaton among population \citep{lively2001trematode} and was able to confirm empirically [CITE]. However, such correlation was not observed in a similar snail-trematode system [CITE] and Lively's original model rather relies on simple assumption.

In this study, we extend the model suggested by \cite{lively2010epidemiological} to include stochasticity and population structure and fit the to data from \cite{dagan2013clonal, mckone2016fine, vergara2014infection} using approximate bayesian computation (ABC).
[Something about what we found out from the model fitting]
Using posterior distributions obtained from the fit, we perform power analysis to test the idea that infection prevalence is positively correlated with frequency of sexual reproduction \cite{lively2001trematode}. 
We find that cycling of parasites and hosts can lead to negative correlation between prevalence of infection and sex. Instead, positive correlated predicted by \cite{lively2001trematode} is driven by populations that do not go through Red Queen dynamics (i.e., having low level of sexual and infection).

\section{Methods}

\subsection{Data}

We consider two snail-trematode population in New Zealand \citep{mckone2016fine, vergara2014infection} and a similar snail-trematode population in Isral \citep{dagan2013clonal}. First, the snail-trematode system in New Zealand has been extensively studied under the Red Queen Hypothesis [CITE] so we expect a Red Queen model to fit well. Second, [Something about Israel snails being different]. Finally, by considering both spatial \citep{dagan2013clonal, mckone2016fine} and spatiotemporal data \citep{vergara2014infection}, we can compare how structure of the data may affect model fits. 
Data for \cite{vergara2014infection} and \cite{dagan2013clonal} were obtained from their Dryad repository [CITE] and data for \cite{mckone2016fine} was extracted from their figure.

\subsection{Model}

We model obligate sexual hosts competing with obligate asexual hosts in multiple population by extending the model introduced by \cite{lively2010epidemiological}.
A model that \cite{lively2010epidemiological} used is essentially an SI model on a discrete time scale with natural mortality and virulence (reduction in offspring production by infected hosts).
It is a suitable canditate for this study as it captures essential structures that are present in basic epidemiological and population population dynamics models and is general enough to be applied to broad systems in nature.
We do not consider mechanistic details of the snail-trematode system such as life history of trematodes \citep{vergara2014infection} as it would require us to estimate or make assumptions on parameters on mechanisms, both of which are unrealistic.

All hosts are assumed to be diploids with two biallelic loci, and parasites are assumed to be haploids.
Let $S_{ij}^k(t)$ and $A_{ij}^k(t)$ be the number of sexual and asexual hosts with genotype $ij$ from population $k$ at generation $t$. 
For simplicity, we drop the superscript representing population and write $S_{ij}(t)$ and $A_{ij}(t)$ unless necessary.
Following \cite{lively2010epidemiological}, the expected number of offsprings (without recombination or outcrossing) produced by a sexual population is given by
\begin{equation}
S_{ij}' = c_b (1-s) \left(W_U S_{ij,U} (t) + W_I S_{ij,I} (t)\right),
\end{equation}
where $s$ is the proportion of males produced, $S_{ij, U}$ and $S_{ij,I}$ are the number of uninfected and infected sexual hosts in a population, and $W_U$ and $W_I$ are their corresponding fitnesses.
We allow for cost of sex to vary by using a scale parameter, $c_b$, to the growth term ($2/c_b$ corresponds to two fold cost of sex) \citep{ashby2015diversity}.
Equivalently, $S_{ij}'$ is the amount of genotypic contribution by sexual hosts to the next generation.

Asexual hosts are assumed to be strictly clonal. Then, the expected number of offsprings produced by an asexual population is given by
\begin{equation}
A_{ij}' = W_U A_{ij,U} (t) + W_I A_{ij,I} (t),
\end{equation}
where $A_{ij, U}$ and $A_{ij,I}$ are the number of uninfected and infected asexual hosts in a population.

We assume that there is mixing between population, determined by rate $\epsilon_{\textrm{mix}}$.
Then, the expected number of offspring in the next generation (accounting for contributions from all populations) is given by
\begin{equation}
\begin{aligned}
\mathrm{E}\left(S_{ij}^k(t+1)\right) &= f_{\textrm{sex}}\left((1 - \epsilon_{\textrm{mix}}) \left(S_{ij}^k\right)' + \frac{\epsilon_{\textrm{mix}}}{n_{\textrm{population}}-1} \sum_{l \neq k} \left(S_{ij}^l\right)'\right),\\
\mathrm{E}\left(A_{ij}^k(t+1)\right) &= (1 - \epsilon_{\textrm{mix}}) \left(A_{ij}^k\right)' + \frac{\epsilon_{\textrm{mix}}}{n_{\textrm{population}}-1} \sum_{l \neq k} \left(A_{ij}^l\right)',\\
\end{aligned}
\end{equation}
where $f_{\textrm{sex}}(x)$ is the function that models sexual reproduction, including recombination at a rate $r_{\textrm{host}}$ and outcrossing, and $n_{\textrm{population}}$ is the number of populations present in the system.
We then take a poisson random variable to simulate process error and allow for stochastic migration to avoid fixation:
\begin{equation}
\begin{aligned}
S_{ij}^k(t+1) &\sim \mathrm{Poisson}\left(\lambda=\mathrm{E}\left(S_{ij}^k(t+1)\right)\right) + \mathrm{Bernoulli}\left(p=p_{ij,\textrm{sex}}\right),\\
A_{ij}^k(t+1) &\sim \mathrm{Poisson}\left(\lambda=\mathrm{E}\left(A_{ij}^k(t+1)\right)\right) + \mathrm{Bernoulli}\left(p=p_{ij, \textrm{asex}}\right),
\end{aligned}
\end{equation}
where $p_{ij,\textrm{sex}}$ and $p_{ij, \textrm{asex}}$ are the probability of a sexual and an asexual host with genotype $ij$ entering the popualation.

Infection is modeled using the matching alleles model \citep{otto1998evolution}.
We assume that heterozygous individuals are equally susceptible to parasites that match either genotypes 

The expected number of infected hosts that carry parasite with genotype $i$ at generation $t$ is given by:
\begin{equation}
I_{i}(t) = \sum_{p} 2^{\delta_{ij}} \left( S_{ip,i,I}(t) + A_{ip,i,I}(t)\right),
\end{equation}
where $S_{ip,i,I}$ and $A_{ip,i,I}$ are the expected number of sexual and asexual hosts infected with genotype $i$ parasite. 
Following \cite{ashby2015diversity}, we assume that mutation can occur in one locus with probability $r_{\textrm{parasite}}$. 
We also allow for external migration of an infected host carrying parasite $i$ with probability $p_{i, \textrm{parasite}}$ to avoid fixation.

The amount of infectious contacts made by infected hosts from a population is given by $\lambda_i^k = \beta^k {I_i'}^k(t)$, where $\beta^K$ is the transmission rate at each population, and $I_i'(t)$ is the number of infected hosts accounting for mutation. 
Since we allow for mixing between population, infected hosts in a population can make contact to susceptible hosts in other populations as well.
Then, all infectious contacts made by infected hosts that applies to a host in a single population is
\begin{equation}
\lambda_{i, \textrm{total}} = (1 - \epsilon_{\textrm{site}}) \lambda_i + \frac{\epsilon_{\textrm{site}}}{n_\textrm{site}-1} \sum_{l \neq k} \lambda_i^l
\end{equation}
Then, the force of infection that a susceptible host experiences is
\begin{equation}
\mathrm{FOI}_{ij}^k = \frac{\lambda_{i, \textrm{total}}  + \lambda_{j, \textrm{total}}  }{2 N(t+1)}
\end{equation}
The probability of infection for a host with genotype $ij$ at site $k$ in the next generation is
\begin{equation}
P_{ij}^k(t+1) = 1 - \exp\left(\mathrm{FOI}_{ij}^k\right).
\end{equation}

Finally, number of infected hosts in the next generation is determined by a binomial process:
\begin{equation}
\begin{aligned}
S_{ij,I} (t+1) &\sim \mathrm{Binom}(S_{ij} (t+1), P_{ij}),\\
A_{ij,I} (t+1) &\sim \mathrm{Binom}(A_{ij} (t+1), P_{ij}).
\end{aligned}
\end{equation}
Expected number of individuals infected with haplotype $i$ in the next generation is given by a ratio of $\lambda$:
\begin{equation}
\begin{aligned}
S_{ij,i,I}(t+1) &= \frac{\lambda_{i, \textrm{total}}}{\lambda_{i, \textrm{total}} + \lambda_{j, \textrm{total}}} S_{ij,I}(t+1)\\
A_{ij,i,I}(t+1) &= \frac{\lambda_{i, \textrm{total}}}{\lambda_{i, \textrm{total}} + \lambda_{j, \textrm{total}}} A_{ij,I}(t+1)
\end{aligned}
\end{equation}

\subsection{Simulation design and parameterization}

Most Red Queen models have studied a competition between a single asexual genotype and sexual genotypes or have assumed equal genetic diversity of asexual and sexual hosts (see \citep{ashby2015diversity}) but neither of these assumptions are realistic.
Instead, \cite{ashby2015diversity} adopted a more realistic approach by allowing for stochastic migration of an asexual genotype to a population.
Here, we combine these methods.
We allow for stochastic migration of migration of asexual hosts with genotypes genotypes but fix number of asexual genotypes (denoted by $G_{\textrm{asex}}$) that can be present in the population, 
while the number of sexual genotypes ($G_{\textrm{sex}}$) that can be present in the population is equal to size of the genotypic space ($=10$ in diploid hosts with two biallelic loci).

In the beginning of every simlation, $G_{\textrm{asex}}$ is randomly chosen between 1 and 10 and asexual genotypes that can be introduced to the population are uniformly sampled from the entire genotypic space.
To account for differing number of sexual and asexual genotypes, we let 
\begin{equation}
\begin{aligned}
p_{ij, \textrm{sex}} &= 1-(1-p_{\textrm{host}})^{1/G_\textrm{sex}},\\
p_{ij, \textrm{asex}} &=
\begin{cases}
1-(1-p_{\textrm{host}})^{1/G_\textrm{asex}} & \text{if } ij \in \{\text{asexual genotypes}\} \\
0 & \text{otherwise}
\end{cases},
\end{aligned}
\end{equation}
where $p_{\textrm{host}}$ is the probability that at least one sexual and asexual host enters the population in a generation. We scale the probability of infected host carrying parsite genotype $i$ in a similar way for interpretability:
\begin{equation}
p_{i, \textrm{parasite}} = 1 - (1-p_{\textrm{infected}})^{1/4},
\end{equation}
where $p_{\textrm{infected}}$ is the probability that at least one infected host enters the population in a generation.

Each simulation consists of 40 populations. Every population consists of 2000 sexual hosts where 80 of them are infected. They are assumed to be in Hardy-Weinberg equilibrium with ratio between alleles being exactly half. Transmission rate parameter, $\beta$, is randomly drawn from a lognormal distribution with parameters $\beta_{\textrm{meanlog}}$ and $\beta_{\textrm{sdlog}}$ for each population. Simulation runs for 500 generations without introduction of asexuals. At generation 501, 10 asexual hosts of a single genotype are introduced to each population (note that asexual genotype introduced can vary across population) and simulation runs for 600 generations while allowing for stochastic migration of asexuals as well.

\subsection{Approximate Bayesian Computation}

We use Approximate Bayesian Computation (ABC) to fit the model \citep{toni2009approximate}.
ABC relies on comparing summary statistics of observed data and those of simulated data and can be particularly useful when the exact likelihood function is not available [CITE?].
We consider mean proportion of infected and sexually reproducing snails in the system and variation in these proportions -- measured by coefficient of variation (CV) -- across space and time as summary statistics of interest.

For each sample from the posterior, the model is simulated and a subsample from the simulated data is drawn so that the number of simulated population is equal to the number of sites collected in a study.
Then, the summary statistics are calculated by considering last 100 generations of simulated data and the parameter sample as well as the simulation generated by the sample is accepted if the distance between simulated and observed data (measureed by the absolute sum of differences in summary statistics) is less than the tolerance value.
By simulating greather number of populations than the observed, we account for uncertainty in unobserved population as well as interaction between populations.

We use uninformative priors for all parameters except $c_b$, a scale parameter for the cost of sex.
The prior distribution for the scale parameter is chosen so that 95\% quantile of cost of sex (2/$c_b$) is approximately equal to the 95\% confidence interval reported by \cite{gibson2017two} (see ??? for the list of parameters).
As \cite{dagan2013clonal} and \cite{mckone2016fine} report proportion of males, instead of proportion of sexual hosts, we double the proportion of males reported when we fit the model.

To allow for more efficient estimation of the posterior distribution, we use Sequential Monte Carlo \cite{turner2012tutorial} (see Appendix for details).

\section{Results}

Our model can reproduce summary statistics observed across different systems reasonably well (\fref{smcsumm}).
Across all systems, most of the observed summary statistics lie within the range of those predicted by the model.
However, our model tends to overestimate mean infection prevalence across all three systems.
In particular, \cite{mckone2016fine} reported mean infection prevalence of approximately 5\% but the model predicts mean infection prevalence of approximately 24\%. %% TODO: change this value once the simulations are over
Mean infection prevalence reported by \cite{mckone2016fine} almost lies outside the range of mean prevalence predicted by the model.
The model also overestimates mean proportion of sexual hosts across population for \cite{vergara2014infection} by 10\% but observed mean proportion still lies within a reasonable range of predicted values.

\begin{figure}[htbp]
\includegraphics[width=\textwidth]{../fig/smc_summary.pdf}
\caption{{\bf Summary statistics of the observed data v.s. simulated data from the posterior samples.}
Violin plots show distribution of summary statistics from each simulated set. For each simulated data set accepted along with the posterior parameter. a set of simulated populations are sampled at random 100 times and summary statistics are calculated for each bootstrap sample by considering last 100 generations. Squares represnt observed summary statistics.
}
\label{fig:smcsumm}
\end{figure}

We also compare mean proportion of sexual hosts against mean proportion of infected hosts (averaged over last 100 generations) of each simulated population along with the observed proportions (\fref{ivs}).
\fref{ivs} may seem to suggest the model fits are much worse overall than what we expected by looking at \fref{smcsumm}.
It is important to note that we are plotting the mean proportions (suppressing the effects of the cycle) whereas observed data is more likely to be traces of the Red Queen cycles.

\begin{figure}[htbp]
\includegraphics[width=\textwidth]{../fig/simulated_data.pdf}
\caption{{\bf Comparison in trends of the observed data v.s. simulated data from the posterior samples.}
Solid circles represent a population within a simulation. Triangles represent observed data.}
\label{fig:ivs}
\end{figure}

The most striking result from \fref{ivs} is that mean proportion of infected hosts decreases or does not change very much while mean proportion of sexual hosts increases in the intermediate range of infection prevalence (most clearly seen from the fits to \cite{vergara2014infection}). %% TODO: add circle around it??
As infection rate increases, selection for sexual hosts increases but increasing number of resistant offsprings prevents further infection from happening.
Our finding contrasts with the work by \cite{lively2001trematode} who predicted positive correlation in prevalence of infection and frequency of sexually reproducing hosts.
We find that proportion of sexual hosts decrease when infection prevalence is very high.
Decrease in fitness of sexual hosts with increase in prevalence was predicted by \cite{ashby2015diversity} but was also observed in an earlier work by \cite{lively2010epidemiological} (although not discussed).

Despite being able to reproduce summary statistics reported by \cite{dagan2013clonal}, 
\fref{ivs} shows that our model cannot capture qualitative relationship bewteen proportion of sexual hosts and proportion of infected hosts.
While both simulated data and observed data mostly consist of asexual populations,
our model predicts some amount of sexual reproduction to be maintained at intermediate infection prevalence. 
However, the data presented by \cite{dagan2013clonal} show that sexual reproduction is only maintained at a low level of infection prevalence and is barely maintained otherwise.
Due to such badness of fit, we exclude fits to \cite{dagan2013clonal} from further discussion.

The model estimates similar parameters for both snail-trematode systems studied by \cite{mckone2016fine} and \cite{vergara2014infection} (\fref{smcparam}).
The most surprising result is that the posterior distribution of the scale parameter, $c_b$, estimated by the model is much wider than the prior distribution.
In particular, the posterior distribution is spread towards the right side of prior distribution and has noticeably higher mean: (mean (95\% credible interval)) 1.18 (0.73-1.79) and 1.24 (0.77-1.80) for \cite{mckone2016fine} and \cite{vergara2014infection}, repsectively. %% TODO: change these once simulations are over! Do we need to consider HPD region? does it matter? 
\cite{ashby2015diversity} defined $c_b$ as additional costs and benefits of sex, where $c_b=1$ corresponds the two fold cost.
Under their interpretation, our estimate corresponds to the following mean and 95\% CIs for cost of sex: 1.81 (1.12-2.74) and 1.68 (1.11-2.59), respectively.

\begin{figure}[htbp]
\includegraphics[width=\textwidth]{../fig/smc_param.pdf}
\caption{{\bf Parameter estimates.}
(dashed lines) prior distribution (solid lines) posterior distribution.
}
\label{fig:smcparam}
\end{figure}

The model estimates high virulence overall and almost 100\% virulence for the system studied by \cite{mckone2016fine}: (mean (95\% CI)) 89\% (75\%-100\%) and 95\% (72\%-100\%) for \cite{mckone2016fine} and \cite{vergara2014infection}, repsectively.
This is consistent with the fact that trematodes are sterilising parasites [CITE].
In contrast to \cite{lively2010epidemiological}, who modeled 1 asexual genotype competing with 9 sexual genotypes, our model estimate shows that higher asexual to sexual genotypic ratio can be supported ($G_{\textrm{asex}}$ panel in \fref{ivs}).

Finally, a power analysis reveals that the power for detecting a positive correlation in proportion of sexual hosts and proportion of infected hosts within each population is not very high (\fref{power}).
Given that \cite{mckone2016fine} collected about 100 snails from each of the 18 sites, they would have had 44\% power to detect the positive correlation. %% TODO: check this value after the simulations are over
Surprisingly, the power to detect positive correlation in the system studied by \cite{vergara2014infection} is extremely low (less than 10\%) and there is much higher power in detecting a negative correlation.
Increasing number of samples per site has small effect on power once the sample size greater than 50.
Increasing number of sites leads to greater increase in power but power appears to saturate.

\begin{figure}[htbp]
\includegraphics[width=\textwidth]{../fig/power.pdf}
\caption{{\bf Power to detect positive correlation.}
Spearman's rank correlation was used.}
\label{fig:power}
\end{figure}

\section{Discussion}

We demonstrated that a simple Red Queen model with population structure is able to produce similar dynamics as those observed in nature and can be used to make inference about natural systems.
Power analysis based on simulated data shows that detecting a positive correlation between infection prevalence and frequency of sexual hosts is difficult.

Our results dramatically differ from the prediction made by \cite{lively2001trematode} and those of many empirical studies that have confirmed the prediction [CITE]. 
An earlier work by \cite{vergara2013geographic} that studied the same population as \cite{vergara2014infection} reported a positive correlation between between infection prevalence and male frequency, contradicting the results from power analysis that predicted almost no power for detecting the positive correlation (\fref{power}).
The difference can easily be explained and call for change in perspective from which Red Queen Hypothesis is studied.

The key premise behind the prediction by \cite{lively2001trematode} is that there must large variation in infection prevalence.
Comparing \cite{vergara2013geographic} and \cite{vergara2014infection} shows that the positive correlation in the population is driven by populations that have low infection prevalence and low frequency of sexual hosts.
It is even clearer in \cite{kumpulainen2004parasites} that positive correlation will not be detected .
Hence, a positive correlation between prevalence of infection and frequency of sexual hosts carries information not about populations that support the Red Queen Hypothesis (goes under parasite mediated selection for sex) but about populations that do not support the Red Queen Hypothesis (does not involve parasites).


While predicting a positive correlation between prevalence of infection and frequency of sexual hosts, \cite{lively2001trematode} also noted that there is a region at which expected infection prevalence overlaps between sexual and asexual populations. 
Within this region lies the association between increase in sexual reproduction and decrease in infection prevalence observed in \fref{ivs}.
Furthermore, the prediction was made by looking at expected mean prevalence of infection and frequency of sexual hosts but does not take temporal effects into account.
A main component of the Red Queen Hypothesis is the cycling between parasite and host populations driven by negative frequency dependence.
When temporal variations are taken into account, association between infection prevalence and frequency of sexual hosts can change depending on where the population is on the cycle and differ from predictions made by looking at mean proportions. %%% make a figure!!!



Discrepancies between the model and the world can ??

High estimate of infection prevalence can be explained by limited genetic diversity of the model.
Previous studies have shown that moderately high genetic diversity can allow sexual hosts to escape infection more easily and therefore reduces prevalence of infection while maintaining sexual reproduction \citep{lively2010effect, king2012does, ashby2015diversity}.
However, our model assumes that infection is deteremined two biallelic loci.
Although not much is known about exact loci that determine trematode infecion in snails, it is not likely that a two loci model is sufficient to capture the genetic diversity of snails [CITE].
Therefore, we expect a model with limited genetic diversity to have higher infection prevalence than observed prevalence.

%% depressing power 
%% way out
%% (1) don't measure correlation look at time series
%% (2) draw a curve?
%% worth noting lively temporal geographic

While all three studies \citep{dagan2013clonal, mckone2016fine, vergara2014infection} look at snail-trematode systems, the system studied by \cite{dagan2013clonal} live in intrinsically different environments.
[TODO: add details]
Our model not take any environmental factors into consideration and only depends on parasite mediated selection for sex.
It is not expected that our model should fit well.

In the absence of the parameter $c_b$ (i.e, when $c_b$ = 1), our model assumes two fold cost of sex in growth rate.
Then, an estimate of $c_b > 1$ can be interpreted as the amount of compensation required in order to overcome the two fold cost of sex.
Additional compensation may include other mechanisms that helps maintenance of sex, such as higher genetic diversity.

Similarly, an estimate of $c_b < 1$ implies that the model can sufficiently produce observed data without any compensation.



\bibliography{redqueen}
\end{document}
