\documentclass{article}\usepackage[]{graphicx}\usepackage[]{color}

\title{Bridging the gap between theory and data: the Red Queen Hypothesis for sex}
\author{Sang Woo Park, Benjamin M Bolker}

\usepackage{tabularx}

\usepackage{amsmath}
\usepackage{natbib}
\usepackage{hyperref}
\bibliographystyle{chicago}
\date{\today}

\usepackage{bm}

\usepackage{afterpage}
\usepackage{pdflscape}

\newcommand{\etal}{\textit{et al.}}

\newcommand{\comment}[3]{\textcolor{#1}{\textbf{[#2: }\textit{#3}\textbf{]}}}
\newcommand{\bmb}[1]{\comment{cyan}{BMB}{#1}}
\newcommand{\swp}[1]{\comment{magenta}{SWP}{#1}}
\newcommand{\citetapos}[1]{\citeauthor{#1}'s \citeyearpar{#1}}

\newcommand{\fref}[1]{Fig.~\ref{fig:#1}}
\begin{document}

\maketitle

\section*{Abstract}

Why does sex continue to persist despite its large cost?
The Red Queen Hypothesis for sex suggests that sexual reproduction is maintained in the host population under strong parasitic pressure by producing genetically rare offspring that are resistant to infection.
Mathematical models have been used to lay theoretical foundations for the hypothesis and empirical data to confirm model predictions.
For example, Lively predicted the frequency of sexual hosts to be positively correlated with the prevalence of infection using a simple host-parasite model and confirmed the prediction through observational studies of snail-trematode systems from New Zealand.
In this study, we fit a simple metapopulation host-parasite coevolution model to observational data from snail-trematode systems and perform a power analysis to test how likely it is to observe the predicted correlation.
We discuss ways in which model prediction differs from observation and provide practical guidance to both modelers and empiricists.
Overall, our study suggests that a simple Red Queen model can only partially explain the observed relationships.

\section{Introduction}

% The evolution of sexual reproduction poses a continuing question \citep{otto2009evolutionary}.
Despite being the dominant mode of reproduction in multicellular organisms \citep{vrijenhoek1998animal}, sexual reproduction entails numerous costs \citep{lehtonen2012many}.
The most commonly mentioned is the cost of producing males \citep{smith1978evolution}:
as males cannot produce offspring, sexual lineages are expected to be outgrown by their asexual counterparts that can grow as twice as fast.
This infamous \emph{two-fold cost of sex} \citep{smith1978evolution} relies on the assumption that everything else is equal.
Then, what else is not equal and drives the sex to persist?

One explanation for the persistence of sexual reproduction is the Red Queen Hypothesis \citep{bell1982masterpiece}.
The Red Queen Hypothesis for sex predicts that sexually reproducing hosts overcome the cost of sex under strong parasitic pressure by producing genetically diverse offspring that can escape infection \citep{jbs1949disease, jaenike1978hypothesis, hamilton1980sex, hamilton1990sexual}.
On the other hand, limited genetic diversity of asexually reproducting hosts incurs a greater fitness cost to infection, allowing for sexual reproduction to be maintained in the host population [CITE].

Much of the theoretical literature has focused on determining qualitative conditions under which parasite selection can maintain sexual reproduction in the host population.
Here, we describe a few important qualitative requirements that must be met in order for parasites to promote sexual reproduction.
First, hosts and parasites must coevolve \citep{bell1982masterpiece}.
Host-parasite coevolution creates a time-lagged selective advantage for rare host genotypes, creating an oscillation in genotypic frequencies and allowing for sexual reproduction to be maintained in the host population \citep{clarke1976ecological, hamilton1980sex};
without persistent coevolution, sexual reproduction can only provide short-term advantage over asexual reproduction [CITE].
Second, parasites must be highly virulent \citep{may1983epidemiology}.
Although sexual and asexual hosts can coexist at intermediate virulence \citep{howard1994parasitism}, sexual reproduction will not provide enough advantage to overcome the cost of sex against avirulent parasites \citep{howard1994parasitism}.
Finally, sexual hosts must be genetically more diverse than asexual hosts, as high clonal diversity may mask the advantage of sexual reproduction \citep{lively2010review, ashby2015diversity}.

Some theoretical studies have departed from the classical population genetics framework to study effects of ecological and epidemiological structures on the Red Queen dynamics [CITE].
A number of studies showed that incorporating ecological and epidemiological details can assist in supporting sexual reproduction in the host population \citep{galvani2001antigenic, galvani2003maintenance, lively2009maintenance, lively2010epidemiological}.
In contrast to these findings, \cite{macpherson2017joint} showed that Red Queen dynamics (i.e., cycles in allele frequencies) fail to persist when explicit epidemiological structure is taken into account with coevolutionary dynamics.
\swp{Transition}

On the other hand, empirical studies have mostly focused on confirming predictions that stem from the Red Queen Hypothesis.
Typical among them are local adaptation, time-lagged selection, and association between parasite prevalence and host reproductive mode (see \cite{tobler2008expanding} and \cite{vergara2014infection} for reviews).
A key example is the snail population in New Zealand that serves as an intermediate hosts for trematodes \citep{winterbourn1974larval, mcarthur1976suppression}.
Through several decades of work, Lively \etal\ have demonstrated that the population satisfies necessary conditions for the host-parasite coevolutionary dynamics and provides support for the hypothesis (e.g., \cite{lively1987evidence, lively1989adaptation, dybdahl1995host, dybdahl1998host, jokela2009maintenance, vergara2014infection, gibson2016within}).
While field studies provide only indirect evidence for the hypothesis, 
recent studies show that more direct evidence can be achieved using experimental systems \citep{auld2016sex, slowinski2016coevolutionary}.

Even though the Red Queen Hypothesis has gained both theoretical and empirical support, there still remains a gap between theory and data.
Many theoretical models rely on simplifying assumptions that are not applicable to natural populations and make indirect connections to empirical studies.
For example, none of the Red Queen models reviewed by \cite{ashby2015diversity} make statistical connections to empirical data.
It is unclear how well these models capture coevolutionary dynamics in natural systems.

However, theoretical models can still be used to make \emph{qualitative} predictions about nature.
\cite{lively1992parthenogenesis} initially postulated that infection prevalence should be positively correlated with frequency of sexual hosts and later formalized the idea using a mathematical model \citep{lively2001trematode}.
The prediction has since been confirmed in several empirical studies, most of which are based on the snail-trematode system \citep{lively2002temporal, kumpulainen2004parasites, king2011parasites, vergara2013geographic, mckone2016fine, gibson2016within}.
Surprisingly, such correlation was not observed in a different snail-trematode system \citep{dagan2013clonal}.

Here, we try to help bridge the gap between theory and data in ... and make quantitative inference about empirical systems.
We extend the model used by \cite{lively2010epidemiological} to account for demographic stochasticity and include simple population structure.
Then, we fit the model to observational data from three studies \citep{dagan2013clonal, mckone2016fine, vergara2014infection} using Approximate Bayesian Computation (ABC) to estimate biologically relevant parameters.
We assess model fits and discuss discrepancies between a theoretical model and the observed data.
Using biologically realistic parameters, we test for power (probability of observing a significant effect) to detect a positive correlation bdetween frequency of sexual reproduction and prevalence of infection \cite{lively2001trematode}.
By performing power analysis, we provide guidance for studying the Red Queen Hypothesis and discuss underlying factors that drive the correlation.

\section{Methods}

\subsection{Data}

\swp{CITE}
We compare three observational data sets \citep{vergara2014infection, mckone2016fine, dagan2013clonal} that represent interactions between freshwater snails and trematodes.
New Zealand snail populations \citep{vergara2014infection, mckone2016fine} consist of \textit{Potamopyrgus antipodarum}, which serves as an intermediate host for several trematode species with \textit{Microphallus} sp. being the most common species.
\cite{vergara2014infection} provide infection status specific to \textit{Microphallus} sp., but \cite{mckone2016fine} did not distinguish among different spcies in their data set.
Israeli snail populations \citep{dagan2013clonal} consist of \textit{Melanoides tuberculata}, which also serves as intermediate host for several trematode species; trematode species are not reported in the data set.
The data sets collected by \cite{dagan2013clonal} and \cite{vergara2014infection} are obtained from their Dryad repositories \citep{dryad_f5t56, dryad_29nk3_2} and the data set collected by \cite{mckone2016fine} is extracted from their figure.

\swp{Write more}
These snail-trematode systems have been extensively studied under the context of the Red Queen Hypothesis so we expect a basic Red Queen model to be able to mimic the observed dynamics reasonably well.
The Israel population is of particular interest as the expected correlation has not been observed consistently \citep{ben2005spatial, dagan2013clonal}.


\subsection{Model}

\swp{TODO: read Lively 2018}
\swp{TODO: also read \citep{schenk2019suicidal}}
We model obligately sexual hosts competing with obligately asexual hosts in a metapopulation by extending the model introduced by \cite{lively2010epidemiological}.
Our model is a discrete time susceptible-infected (SI) model with natural mortality and virulence (defined as reduction in offspring production among infected hosts).
Metapopulation structure is included to model unobserved dynamics among different habitats; equivalently, each population can be considered as a sampling site.
A similar metapopulation model was developed by \cite{lively2017habitat} in order to study local adaptaion of parasites.

We do not model the life history of snail-trematode interactions explicitly.
Modeling interactions of multiple trematodes species, each of which go through a different life cycle, with a single snail species is extremly complicated.
Instead, testing a basic model against real data will allow us to identify the difference between theory and data more clearly.

All hosts are assumed to be diploids with two biallelic loci controlling resistance, and parasites are assumed to be haploids.
Let $S_{ij}^k(t)$ and $A_{ij}^k(t)$ be the number of sexual and asexual hosts with genotype $ij$ from population $k$ at generation $t$, 
where each subscript, $i$ and $j$, represents a host hapolotype: $i, j \in \{\mathrm{AB}, \mathrm{Ab}, \mathrm{aB}, \mathrm{ab}\}$.
For simplicity, we drop the superscript representing population and write $S_{ij}(t)$ and $A_{ij}(t)$;
every population is governed by the same set of equations unless noted otherwise (e.g., when we account for interaction between populations).
Following \cite{lively2010epidemiological}, the expected genotypic contribution (before recombination or outcrossing) by sexual hosts is given by
\begin{equation}
S_{ij}' = c_b (1-s) \bigg(W_U S_{ij,U} (t) + W_I S_{ij,I} (t)\bigg),
\end{equation}
where $s$ is the proportion of males produced by sexual hosts, and $S_{ij, U}$ and $S_{ij,I}$ are the number of uninfected and infected sexual hosts in a population.
$W_U$ and $W_I$ represent their corresponding fitnesses where virulence is defined as $V = 1-W_I/W_U$.
We allow for the cost of sex to vary by multiplying a scale parameter, $c_b$, to the growth term, where $2/c_b$ corresponds to a two fold cost of sex when $c_b = 1$ \citep{ashby2015diversity}.
Recombination and outcrossing are modeled after incorporating genotypic contributions from other populations.

We define
$$
W_U = \frac{b_U}{1 + a_U N(t)},\,  W_I = \frac{b_I}{1 + a_I N(t)}
$$
where $b_U$ and $b_I$ are number of offspring produced by uninfected and infected hosts, respectively, and $a_U$ and $a_I$ determine their corresponding strengths of density dependence \citep{lively2010epidemiological, smith1973stability}.
For simplicity, we assume that $a_U = a_I$ so that virulence can be defined strictly in terms of decrease in offspring production and is density independent: $V = 1- b_I/b_U$.

Asexual hosts are assumed to be strictly clonal.
Then, the expected genotypic contribution by asexual hosts is given by
\begin{equation}
A_{ij}' = W_U A_{ij,U} (t) + W_I A_{ij,I} (t),
\end{equation}
where $A_{ij, U}$ and $A_{ij,I}$ are the number of uninfected and infected asexual hosts in a population.

We assume that a proportion $\epsilon_{\textrm{\tiny mix}}$ of a population mixes with other populations.
Then, the expected number of offspring in the next generation (accounting for contributions from all populations) is given by
\begin{equation}
\begin{aligned}
\mathrm{E}\left(S_{ij}^k(t+1)\right) &= f_{\textrm{\tiny sex}}\left((1 - \epsilon_{\textrm{\tiny mix}}) \left(S_{ij}^k\right)' + \frac{\epsilon_{\textrm{\tiny mix}}}{n_{\textrm{\tiny pop}}-1} \sum_{h \neq k} \left(S_{ij}^h\right)'\right),\\
\mathrm{E}\left(A_{ij}^k(t+1)\right) &= (1 - \epsilon_{\textrm{\tiny mix}}) \left(A_{ij}^k\right)' + \frac{\epsilon_{\textrm{\tiny mix}}}{n_{\textrm{\tiny pop}}-1} \sum_{h \neq k} \left(A_{ij}^h\right)',\\
\end{aligned}
\end{equation}
where $f_{\textrm{\tiny sex}}(x)$ is the function that models sexual reproduction, including recombination probability $r_{\textrm{\tiny host}}$ and outcrossing, and $n_{\textrm{\tiny pop}}$ is the number of populations modeled.
Then, the total number of sexual and asexual hosts in the next generation given by Poisson random variables with mean specified previously. We also allow for stochastic migration in order to avoid fixation:
\begin{equation}
\begin{aligned}
S_{ij}^k(t+1) &\sim \mathrm{Poisson}\left(\lambda=\mathrm{E}\left(S_{ij}^k(t+1)\right)\right) + \mathrm{Bernoulli}\left(p=p_{ij,\textrm{\tiny sex}}\right),\\
A_{ij}^k(t+1) &\sim \mathrm{Poisson}\left(\lambda=\mathrm{E}\left(A_{ij}^k(t+1)\right)\right) + \mathrm{Bernoulli}\left(p=p_{ij, \textrm{\tiny asex}}\right),
\end{aligned}
\end{equation}
where $p_{ij,\textrm{\tiny sex}}$ and $p_{ij, \textrm{\tiny asex}}$ are the probabilities of a sexual and an asexual host with genotype $ij$ entering a popualation.

Infection is modeled using the matching alleles model \citep{otto1998evolution}.
We assume that snails are equally susceptible to parasites that match either haplotype.
However, parasites must match the host haplotype at both loci in order to successfully infect a host.
The expected number of infected hosts that is infected with parasite with genotype $i$ at generation $t$ is given by:
\begin{equation}
I_{i}(t) = f_{\textrm{\tiny mutation}} \left( \sum_{j}  \bigg( S_{ij,i,I}(t) + A_{ij,i,I}(t)\bigg)\right).
\end{equation}
$S_{ij,i,I}(t)$ and $A_{ij,i,I}(t)$ represent the expected numbers of sexual and asexual hosts that have genotype $ij$ and are infected with parasites that have genotype $i$.
The function $f_{\textrm{\tiny mutation}}$ models mutation at a single locus with with probability $r_{\textrm{\tiny parasite}}$ \citep{ashby2015diversity}.

The total expected number of infectious contacts made by infected hosts within a population is given by $\lambda_i^k = \beta^k {I_i'}^k(t)$, where $\beta^k$ is the transmission rate of each population, and $I_i'(t)$ is the number of infected hosts accounting for stochastic migration with probability $p_{i, \textrm{\tiny parasite}}$ to avoid fixation. 
Since we allow for mixing between populations, infected hosts can make contact with susceptible hosts in other populations. \swp{TODO: explain mixing}
Then, the total amount of infectious contact, coming from hosts that carry genotype $i$ parasite, that is received by susceptible hosts in population $k$ is given by
\begin{equation}
\lambda_{i, \textrm{\tiny total}}^k = (1 - \epsilon_{\textrm{\tiny mix}}) \lambda_i^k + \frac{\epsilon_{\textrm{\tiny mix}}}{n_\textrm{\tiny pop}-1} \sum_{l \neq k} \lambda_i^l
\end{equation}
Then, the force of infection that a susceptible host with genotype $ij$ experiences in generation $t+1$ is given by
\begin{equation}
\mathrm{FOI}_{ij}^k = \frac{\lambda_{i, \textrm{\tiny total}}^k  + \lambda_{j, \textrm{\tiny total}}^k}{2 N^k(t+1)},
\end{equation}
where $N^k(t+1) = \sum_{i,j} S_{ij}^k(t+1) + A_{ij}^k(t+1)$ is the total number of hosts in generation $t+1$.
The probability that a susceptible host with genotype $ij$ in population $k$ becomes infected in the next generation is given by
\begin{equation}
P_{ij}^k(t+1) = 1 - \exp\left(\mathrm{FOI}_{ij}^k\right).
\end{equation}

Finally, the number of infected hosts in the next generation is determined by a binomial process:
\begin{equation}
\begin{aligned}
S_{ij,I}^k (t+1) &\sim \mathrm{Binom}\left(S_{ij}^k (t+1), P_{ij}^k(t+1)\right),\\
A_{ij,I}^k (t+1) &\sim \mathrm{Binom}\left(A_{ij}^k (t+1), P_{ij}^k(t+1)\right).
\end{aligned}
\end{equation}
The expected number of infected hosts that have genotype $ij$ and are infected by parasites with genotype $i$ in the next generation is proportional to the amount of infectious contact that was made in the current generation:
\begin{equation}
\begin{aligned}
S_{ij,i,I}^k(t+1) &=  \frac{2^{\delta_{ij}} \lambda_{i, \textrm{\tiny total}}^k}{\lambda_{i, \textrm{\tiny total}}^k + \lambda_{j, \textrm{\tiny total}}^k} S_{ij,I}^k(t+1)\\
A_{ij,i,I}^k(t+1) &=  \frac{2^{\delta_{ij}} \lambda_{i, \textrm{\tiny total}}^k}{\lambda_{i, \textrm{\tiny total}}^k + \lambda_{j, \textrm{\tiny total}}^k} A_{ij,I}^k(t+1)
\end{aligned}
\end{equation}
where a $\delta_{ij}$ is a Kronecker delta ($\delta_{ij}$ equals 1 when $i = j$ and 0 otherwise).

\subsection{Simulation design and parameterization}

Many Red Queen models have focused on competition between a single asexual genotype and multiple sexual genotypes or have assumed equal genetic diversity between asexual and sexual hosts (see \citep{ashby2015diversity} for a review of previous Red Queen models) but neither of these assumptions is realistic. \swp{maybe cite Schenk et al}
Instead, \cite{ashby2015diversity} adopted a more realistic approach by incorporating stochastic external migration of an asexual genotype to a population and allowing for asexual genetic diversity to vary over time.
Here, we combine these methods.
We allow for stochastic external migration of asexual hosts with different genotypes into the system but fix the number of asexual genotypes (denoted by $G_{\textrm{\tiny asex}}$) that can be present in the system. 
Since sexual hosts have limited genetic diversity in our model (diploid hosts with two biallelic loci yields a total of 10 genotypes), allowing for unlimited migration of asexual hosts will cause the sexual population to be easily outcompeted by the asexual population.
Limiting asexual genetic diversity allows us to account for the intrinsic difference in sexual and asexual diveristy and make a compromise between simple and realistic models.
At the beginning of each simulation, a pool of $G_{\textrm{\tiny asex}}$ asexual genotypes are sampled at random the entire genotypic space; these are the genotypes that are available for external immigration into subpopulations.
Then, we use our model-based comparison with data to estimate $G_{\textrm{\tiny asex}}$ to test whether greater asexual genetic diversity can be supported than a typically assumed one-to-many ratio.

To account for differing number of sexual and asexual genotypes, we let 
\begin{equation}
\begin{aligned}
p_{ij, \textrm{\tiny sex}} &= 1-(1-p_{\textrm{\tiny host}})^{1/G_\textrm{\tiny sex}},\\
p_{ij, \textrm{\tiny asex}} &=
\begin{cases}
1-(1-p_{\textrm{\tiny host}})^{1/G_\textrm{\tiny asex}} & \text{if } ij \in \{\text{asexual genotypes}\} \\
0 & \text{otherwise}
\end{cases},
\end{aligned}
\end{equation}
where $p_{\textrm{\tiny host}}$ is the probability that at least one sexual or asexual host enters the population in a generation. We scale the probability of an infected host carrying parasite genotype $i$ in a similar way for interpretability:
\begin{equation}
p_{i, \textrm{\tiny parasite}} = 1 - (1-p_{\textrm{\tiny infected}})^{1/4},
\end{equation}
where $p_{\textrm{\tiny infected}}$ is the probability that at least one infected host enters the population in a generation.
\swp{explain 4; haploid parasites}

Each simulation consists of 40 subpopulations. Every subpopulation is initialized with 2000 sexual hosts, of which 80 are infected. 
They are assumed to be in Hardy-Weinberg equilibrium where allele frequencies in each locus is half. 
The transmission rate, $\beta^k$, is randomly drawn for each subpopulation from a Gamma distribution with mean $\beta_{\textrm{\tiny mean}}$ and coefficient of variation $\beta_{\textrm{\tiny CV}}$. 
Simulation runs for 500 generations without introduction of asexuals. At generation 501, 10 asexual hosts of a single genotype are introduced to each population (the asexual genotype introduced can vary across population) and simulation runs for 600 generations while allowing for stochastic migration of asexuals.

\subsection{Approximate Bayesian Computation}

\bmb{More on probe matching; Kendall et al.?}
\swp{Not clear what you're looking for...}
\swp{TODO: come back again... skipping to results for now}

We use Approximate Bayesian Computation (ABC) to estimate parameters from data \citep{toni2009approximate}.
ABC relies on comparing summary statistics of observed data and those of simulated data and is particularly useful when the exact likelihood function is not available.
We consider mean proportion of infected and sexually reproducing snails in the system and variation in these proportions -- measured by coefficient of variation (CV) -- across space (population) and time as our focal summary statistics.
These summary statistics are calculated for both observed and simulated data and are used in ABC.
As \cite{dagan2013clonal} and \cite{mckone2016fine} only reported the proportion of males, the proportion of sexual hosts is assumed to be twice the proportion of males.

CV across space is calculated by first calculating mean proportions by averaging across time (generation) for each site (subpopulation) and then taking the the CV of these mean proportions.
CV across time (generation) is calculated by first averaging proportions across space (subpopulation) at each generation and then taking the CV.
For purely spatial data (\cite{dagan2013clonal} and \cite{mckone2016fine}), CV across space is calculated without averaging across time.

Because ABC is a Bayesian method, we want to specify prior distributions for all parameters.
We use weakly informative priors for all parameters that we estimate except $c_b$, a scale parameter for the cost of sex (see Table~\ref{tb:param} for prior distributions used and parameters assumed).
The prior distribution for the scale parameter is chosen so that 95\% prior quantile of cost of sex (2/$c_b$) is approximately equal to the 95\% confidence interval reported by \cite{gibson2017two}.
All other parameters are assumed to be fixed.



\afterpage{
\clearpage
\begin{landscape}
\begin{table}[h]
\centering
\begin{tabular}{c|p{5cm}|c|c}
\hline
\textbf{Notation} & \textbf{Description} & \textbf{Prior distribution/parameter values} & \textbf{Source}\\
\hline
$\beta_{\textrm{\tiny mean}}$ & Mean transmission rate & $\mathrm{Gamma}(k=2, \theta=10)$ & Assumption\\
$\beta_{\textrm{\tiny CV}}$ & CV transmission rate & $\mathrm{Gamma}(k=2, \theta=0.5)$ & Assumption\\
$V$ & Virulence & $\mathrm{Beta}(\alpha=6, \beta=2)$ & Assumption\\
$\epsilon_{\textrm{\tiny mix}}$ & Mixing proportion & $\mathrm{Beta}(\alpha=1, \beta=9)$ & Assumption\\
$G_{\textrm{\tiny asex}}-1$ & Number of asexual genotypes - 1 & $\mathrm{BetaBinomial}(N=9, p=3/9, \theta=5)$ & Assumption\\
$c_b$ & Cost of sex scale & $\mathrm{LogNormal}(\mu=-0.07, \sigma=0.09)$ & \cite{gibson2017two}\\
$s$ & Proportion of male offsprings produced & 0.5 & Assumption\\
$b_U$ & Number of offsprings produced by an uninfected host & 20 & \cite{lively2010epidemiological}\\
$b_I$ & Number of offsprings produced by an infected host & $(1-V) b_U$ & \cite{lively2010epidemiological}\\
$a_U$ & Density dependent effect coefficient of uninfected hosts & 0.001 & \cite{lively2010epidemiological}\\
$a_U$ & Density dependent effect coefficient of infected hosts & 0.001 & \cite{lively2010epidemiological}\\
$r_{\textrm{\tiny host}}$ & Host recombination probability & 0.2 & \cite{lively2010epidemiological}\\
$r_{\textrm{\tiny parasite}}$ & Parasite mutation probability & 0.05 & Assumption\\
$p_{\textrm{\tiny host}}$ & Probability that at least one sexual and asexual host enters the population & 0.1 & Assumption\\
$p_{\textrm{\tiny infected}}$ & Probability that at least one infected host enters the population & 0.02 & Assumption\\
\hline
\end{tabular}
\caption{
\textbf{Parameter descriptions and values}.
Parameters with prior distributions are estimated via Approximate Bayesian Computation (ABC).
$k$ and $\theta$ in Gamma distribution represent shape and scale parameters where mean and squared CV are given by $k \theta$ and $1/k$, respectively.
$\alpha$ and $\beta$ in Beta distribution represent shape parameters where mean and squared CV are given by $\alpha/(\alpha+\beta)$ and $\beta/(\alpha^2 + \alpha \beta + \alpha)$.
$N$, $p$ and $\theta$ in Beta binomial distributions represent number of trials, probability of success, and overdispersion parameters \citep{morris1983natural}.
We define prior on $G_{\textrm{\tiny asex}} -1$ instead to always maintain at least one asexual genotype in the system.
$\mu$ and $\sigma$ in log-normal distribution represent mean and standard deviation on a log scale.
All other parameters are fixed throughout simulations.
}
\label{tb:param}
\end{table}
\end{landscape}
}

% To allow for more efficient estimation of the posterior distribution, we use the Population Monte Carlo approach \citep{turner2012tutorial}.
We start by performing basic ABC.
For each random parameter sample drawn from the prior distribution, the model is simulated and a sample of simulated populations is drawn from the simulated system such that the number of sampled population is equal to the number of sites collected in a study.
Then, summary statistics are calculated based on the last 100 generations out of 1100 generations and the parameter is accepted if the distance between simulated and observed data is less than a tolerance value.
Distance is measured by the sum of absolute differences in summary statistics between simulated and observed data.
This process is repeated until 100 parameter sets are accepted.

After the first run ($t=1$), equal weights ($w_{i,1}=1/100$) are assigned to each accepted parameter set $\bm\theta_{i, 1}$, where $1 \leq i \leq 100$.
For any run $t > 1$,
a weighted random sample ($\bm\theta^\ast$) is drawn from the accepted parameters of the previous run ($t-1$) with weights $w_{i,t-1}$ and
a parameter sample ($\bm\theta_{i, t}$) is proposed from a multivariate normal distribution with a mean $\bm\theta^\ast$ and a variance covariance matrix that is equal to $\sigma_{t-1}^2=2 \mathrm{Var}(\bm\theta_{1:N, t-1})$, where $\mathrm{Var}(\bm\theta_{1:N, t-1})$ is the weighted variance covariance matrix of the accepted parameters from the previous run.
$N$ is the total number of accepted parameters from the previous run.

$G_{\textrm{\tiny asex}}$ is rounded to the nearest integer and the model is simulated.
If a proposed parameter is accepted, the following weight is assigned:
$$
w_{i,t} = \frac{\pi(\bm\theta_{i, t})}{\sum_{i=1}^{100} w_{j, t-1} q(\bm\theta_{j, t-1} | \bm\theta_{i,t}, \sigma_{t-1}^2)}
$$
where $\pi(\cdot)$ is a prior density and $q(\cdot | \bm\theta_{i,t}, \sigma_{t-1}^2)$ is a multivariate normal density with mean $\bm\theta_{i,t}$ and variance covariance matrix $\sigma_{t-1}^2$.
For each run, 100 parameters are accepted and weights are normalized at the end to sum to 1.
This method, known as the Population Monte Carlo approach \citep{turner2012tutorial}, allows for sampling more efficiently while ensuring that final result still satisfies criteria to be a correct (approximate) Bayesian posterior.
All statistical results reported are weighted by parameter weights of the final run.

For each observed datum, we perform 4 runs with decreasing tolerance every run.
For spatial data \citep{dagan2013clonal, mckone2016fine}, four summary statistics are compared: mean proportion of infected and sexually reproducing snails and CV in these proportions across populations.
Tolerance values of 1.6, 0.8, 0.6 and 0.4 are used for each run.
For spatiotemporal data \citep{vergara2014infection}, six summary statistics are compared: mean proportion of infected and sexually reproducing snails, CV in these proportions across populations and CV in these proportions across generations.
Larger tolerance values (2.4, 1.2, 0.9 and 0.6) are used for each run to account for higher number of summary statistics being compared.
Tolerance value of the final run is chosen so that a parameter set will be accepted if its each simulated summary statistic deviates from the corresponding observed summary statistic by 0.1 units on average.
First three tolerance values are chosen in a decreasing order to reach the final step quicker.

\subsection{Power analysis}

\bmb{This is a post hoc power analysis. Need to recognize/admit its limitations. Power strongly correlated with significant correlation in original study; not as bad as direct post hoc power analysis but...}

Using estimated parameters for each data set, we calculate the power to detect a significant positive correlation between infection prevalence and frequency of sexual hosts.
For each parameter sample from the final run of the ABC, 10 simulations are run.
For each simulation, we take the last two generations -- assuming that a year contains two snail generations [CITE] -- from the simulation and choose $n$ populations at random from 40 simulated populations.
For each selected population, hosts are divided into four categories based on their infection status (infected/uninfected) and reproductive mode (asexual/sexual),
and mean proportion of hosts in each category is calculated by averaging over two generations.
Independent multinomial samples of size $m$ are drawn from each selected population based on the proportions  in each four categories. 
Correlation between proportion of infected hosts and proportion of sexual hosts is tested using the Spearman's rank correlation at a 5\% significance level.

\section{Results}

First, we compare observed summary statistics with fitted and predicted summary statistics (\fref{smcsumm}).
Fitted summary statistics are those that are accepted via ABC;
these can be interpreted as model-based estimates of the true summary statistics (and associated confidence intervals) of the study sites.
Predicted summary statistics are obtained by simulating the model again using the estimated parameters and calculating summary statistics from randomly selected subpopulations;
these values represent summary statistics that could have been obtained if other sites were chosen for the study.
The predicted summary statistics are highly variable because they account for uncertainty in the unobserved subpopulations.

Our simple meta-population Red Queen model can capture observed variation in infection prevalence and frequency of sexual hosts reasonably well;
both temporal and spatial variation (measured by CV across mean proportions) are well-matched by the model. 
However, as model fitting is performed by minimizing the sum of absoute distance between observed and simulated summary statistics, our method does not gurantee that all summary statistics are equally well-fitted.
The model tends to overestimate mean proportion of infected hosts.
The observed mean proportion of infected hosts are 0.175 \citep{dagan2013clonal}, 0.051 \citep{mckone2016fine}, and 0.440 \citep{vergara2014infection}, 
whereas their ABC-based estimates are 0.240 (95\% CI: 0.174 - 0.286), 0.313 (95\% CI: 0.233 - 0.404), and 0.542 (95\% CI: 0.360 - 0.730), respectively.
The model underestimates mean proportion of sexual hosts for \cite{dagan2013clonal} and \cite{vergara2014infection}.
Observed mean proportions of sexual hosts are 0.045 and 0.704, respectively, whereas their ABC-based estimates are 0.026 (95\% CI: 0.007 - 0.048) and 0.596 (95\% CI: 0.441 - 0.679).
\swp{44.1\% is actually 2.82\% quantile rather than 2.5\%. Should I write a sentence about it? It feels just a little gratuitous, although I think it is better to be clear and honest.}

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/smc_summary.pdf}
\caption{{\bf Summary statistics of the observed data vs. distribution of summary statistics of the simulated data from the posterior samples.}
Dotted horizontal lines represent observed summary statistics.
Violin plots show weighted distribution of fitted summary statistics (i.e., summary statistics that were accepted during ABC). 
Error bars show 95\% weighted quantiles of predicted summary statistics.
The weights correspond to the posterior distribution weights from ABC.
For each posterior sample, 10 simulations are run and each simulated system is sampled at random 100 times so that each sample consists of the same number of populations as in the fitted data.
All univariate summary statistics are matched reasonably well, except for the mean proportion of infected hosts in \cite{mckone2016fine}.
}
\label{fig:smcsumm}
\end{figure}

\swp{updated text and fig:}
To further diagnose the fit, we compare the predicted relationships between the mean proportion of infected hosts and the mean proportion of sexual hosts across subpopulations with the observed data (\fref{ivs});
% we consider infection prevalence and frequency of sexual hosts averaged over the last two generations to remain consistent with power analyses.
Despite its accuracy in reproducing summary statistics reported by \cite{dagan2013clonal}, 
our model poorly captures the relationship between the mean proportion of infected hosts and the mean proportion of sexual hosts (\fref{ivs}; \cite{dagan2013clonal}).
The model predicts sexual reproduction to be well maintained when infection prevalence is high ($> 40\%$) whereas
the observed data \citep{dagan2013clonal} suggests that sexual reproduction can only be supported when infection prevalence is low ($< 20\%$).
\cite{mckone2016fine} also found sexually reproducing snails in sites with low infection prevalence ($< 20\%$).

On the other hand, \citetapos{vergara2014infection} data set suggests that sexually reproducing populations are likely to have a relatively high prevalence of infection (approximately 20\% - 80\%).
This is broadly consistent with our model prediction. 
Most of the data points fall within the range of model predictions (\fref{ivs}; \cite{vergara2014infection}).
However, there is one site in which more than 90\% of the snails were found to be sexual throughout the study period of 5 years \citep{vergara2014infection};

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/simulated_data2.pdf}
\caption{{\bf Predicted relationship between mean infection prevalence and mean proportion of sexual hosts in each population.}
For each posterior sample, 10 simulations are run.
For each population within a simulation, mean infection prevalence and mean proportion of sexual hosts is calculated by averaging across last two generations. 
Each point, consisting of mean infection prevalence and meal proportion of sexual hosts, is assigned the weight of the parameter used to simulated the population.
The density at each cell is calculated as the sum of weights of the points that lie within it.
Cell densities are normalized by dividing by the maximum grid densities for each fit.
\swp{I tweaked color gradient schemes and plotted log relative densities. Need to clarify this?}
Open triangles represent observed data; proportion of sexual hosts is computed from proportion of male hosts.
}
\label{fig:ivs}
\end{figure}

There is a high posterior density region (around 50\% infection prevalence) in which the proportion of infected hosts remains almost constant while the proportion of sexual hosts increases (visible in the fits to \cite{mckone2016fine} and \cite{vergara2014infection}). \swp{Appendix}
As transmission rate ($\beta$) increases, selection for sexual hosts increases but the increasing number of resistant offspring prevents further infection from occuring and can decrease overall infection prevalence.
Such a trend is consistent with previous results of \cite{lively2001trematode} who noted that there is a region in which either sexual and asexual reproduction can be selected exclusively under same infection prevalence.
The proportionf of sexual hosts decreases when infection prevalence is extremely high.
This pattern can be explained by the decrease in fitness of sexual hosts with increase in infection prevalence, predicted by \cite{ashby2015diversity}. 
The same pattern can be found in an earlier work by \cite{lively2010epidemiological} although it was not discussed.

\fref{smcparam} presents parameter estimates.
Keeping in mind that we do not obtain good fits to data from \cite{dagan2013clonal} and \cite{mckone2016fine}, we still find that high virulence and a low ratio of asexual to sexual genetic diversity are necesary to explain the observed dynamics.
Moreover, we are able to capture observed differences in mean and variation in infection prevalence among studies in our estimates of transmission rate parameters ($\beta_{\textrm{\tiny mean}}$ and $\beta_{\textrm{\tiny CV}}$).

Our fits to \cite{mckone2016fine} suggest that scale parameter for the cost of sex, $c_b$, should be higher than our prior assumption based on \cite{gibson2017two} that the estimated cost of sex to be 2.14 (95\% CI: 1.81 - 2.55).
\cite{ashby2015diversity} defined $c_b$ as additional costs and benefits of sex, where $c_b=1$ corresponds the two fold cost.
Under their interpretation, our estimate of $c_b$ corresponds to a slightly lower estimate of cost of sex: 1.95 (95\% CI: 1.68 - 2.4). 
(We propose an alternate interpretation to this parameter estimate below).

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/posterior.pdf}
\caption{{\bf Parameter estimates from Sequential Monte Carlo Approximate Bayesian Computation.}
Violin plots represent weighted distribution of 100 posterior samples obtained from ABC.
Violin plots for prior distribution is obtained by drawing 10000 random parameter samples from the prior distribution.
$G_{\tiny \textrm{asex}}$ is a discrete variable but is drawn on a continuous scale for convenience.
}
\label{fig:smcparam}
\end{figure}

Finally, our power analyses suggest that there is high power to detect a positive correlation between infection prevalence and frequency of sexual hosts in the systems studied by \cite{dagan2013clonal} and \cite{mckone2016fine} (\fref{power}).
Such high power predicted for \cite{dagan2013clonal} is particularly surprising given that they were not able to observe the expected correlation.
This discrepancy implies that the snail populations studied by \cite{dagan2013clonal} show sufficient variation in infection prevalence in order for the correlation to be observed under pure Red queen selection, but other underlying factors that are neglected by our model does not account may have caused the populations to deviate from their expected behaviours.
On the other hand, our model predicts low power for detecting the positive correlation for the system studied by \cite{vergara2014infection} (\fref{smcparam}).
Overall, we predict that increasing number of sites is a more effective way to increase power than increasing number of samples per site.

While we originally planned to perform power analysis using Spearman's rank correlation,
we repeated the analysis using Pearson's correlation after applying arcsine square root transformation [CITE] to see whether this procedure improves power.
Surprisingly, using Pearson correlation gives slightly higher power to detect the positive correlation bewteen frequency of sexual hosts and infection prevalence (see Appendix).
\bmb{Not surprinsing; pearson has higher power if pattern is actually linear}

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/power.pdf}
\caption{{\bf Power to detect a statistically significant positive correlation between infection prevalence and frequency of sexual hosts.}
Spearman's rank correlation was used to test for correlation between infection prevalence and frequency of sexual hosts in simulated data from the posterior distributions.
}
\label{fig:power}
\end{figure}

\section{Discussion}

Our study questions ways in which the Red Queen Hypothesis for sex has been modeled.
Many modeling studies have relied on assumed parameter values to understand the role of host parasite coevolution in maintaining sexual reproduction; 
such method allows us to learn about the model but not so much about the nature.
Instead, we tried to fit a simple Red Queen model to observational data from three different snail trematode systems.
We show that (1) model parameters can be estimated from data and (2) biologically meaningful predictions can be made from the model.
However, discrepancy between model prediction and observed data suggests that a simple host-parasite coevolution model cannot sufficiently explain maintanence of sexual reproduction observed in snail populations.

A model that does not fit well can sometimes tell us more about a biological system than a model that fits well.
For example, there was a clear mismatch between the model prediction and the data presented by \cite{dagan2013clonal} (\fref{ivs}).
The snail populations studied by \cite{dagan2013clonal} live in intrinsically different environments from two other snail populations that we considered.
For example, habitats are subject to seasonal flash floods, which can affect reproductive strategies of snails \citep{ben2007temporal} and interfere with the host parasite coevolution.
As a result, positive correlation between infection prevalence and frequency of sexual reproduction could not be detected from the system even though high power is predicted.
We caution against performing statistical tests that were purely designed under the Red Queen Hypothesis when other mechanisms that may affect host reproductive mode are present in the system.
Moreover, even if the effect of Red Queen cannot be clearly detected, one cannot conclude that there is \emph{no effect} of parasite on maintenance of sex. \swp{Can I cite something here?}

The model fit to \cite{mckone2016fine} suggests that cost of sex can be overcome and sexual reproduction can be maintained only if infection prevalence is much higher than the observed prevalence (\fref{ivs}).
In other words, benefit of producing offspring with novel genotype is relatively small when infection prevalence is low.
Benefit of sex must be greater or other mechanisms must compensate for the difference in order to support sexual reproduction at lower infection prevalence.
As our model relies on a simple structure and strong parametric assumptions, additional benefit of sex can only be provided by lowering the cost of sex (i.e., increasing the scale parameter, $c_b$).

The simple structure of the model and limited genetic diversity provides an explanation for the discrepancy observed in model prediction and the observed data by \cite{mckone2016fine}.
Here, we assumed that host resistance to infection is deteremined entirely by two biallelic loci, which result in 10 genotypes, but it is unlikely that such simple model can capture genetic interaction between hosts and parasites observed in nature.
Although exact genetic architecture that determines trematode infecion in snails (e.g., loci involved in parasite resistance) is not known \swp{How do I even cite this?}, genetic diversity of snails that have been documented is far greater than what we have assumed \citep{king2011parasites, dagan2013clonal}.
Increasing genetic diversity of the model would have allowed sexual hosts to escape infection more easily and maintained sexual reproduction at lower prevalence of infection \citep{lively2010effect, king2012does, ashby2015diversity}.

While the positive correlation between frequency of sexual hosts and prevalence of infection can provide evidence for the effect of parasite on the maintence of sexual reproduction in the host population, it does not capture sufficient information about the evolutionary process.
Precisely, the positive correlation represents contrast between populations that undergo Red Queen dynamics and those that do not; it emphasizes asexual selection due to lack of parasitic risk \citep{lively2001trematode}.
While the similarity between the correlations predicted by the model and the correlations observed in nature may be interpreted as providing more confidence that the observed popoulations provide evidence for the Red Queen Hypothesis (\fref{effect}), the strength of correlation, does not provide any evidence for the strength of selection imposed on the host population.
Moreover, when populations are going through active coevolution, wide range of correlations can be detected (\fref{effect}: \cite{vergara2014infection}).
While simple statistical models such as testing for correlation can be tested easily,
using more sophisticated and mechanistic statistical models that yield biologically interpretable parameters may be better suited for understanding the effect of parasites on the maintenance of sexual reproduction in the host population.
\swp{I realized that Lively provides similar explanations for these phenomena. Is it OK to keep this in the discussion (I do cite him)? Even though the explanations overlap, I think the point is clearer with a slightly more realistic model and is worth keeping?}

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/effect_size.pdf}
\caption{{\bf Predicted vs. observed correlation between infection prevalence and frequency of sexual hosts.}
Violin plots show weighted distribution of predicted strenght of correlation.
Predicted vorrelation was measured for each simulation from the posterior by taking into account last two generations from each simulated population.
Triangles and circles represent observed correlation sizes from previous studies that found positive correlation.
}
\label{fig:effect}
\end{figure}

Mathematical models have used extensively to build theoretical foundations for the evolution of sex but only a few models have been statistically connected to data.
Only recently, the idea of two fold cost of sex was tested directly by fitting a theoretical model to observed data \citep{gibson2017two}.
Making statistical inference on the observed systems and testing theoretical models against data may provide deeper insight into underpinnings of maintenance of sexual reproduction.

\section{Acknowledgements}

We thank SHARCnet for providing computational resources.

\bibliography{redqueen}

\pagebreak
\section*{Appendix}

\renewcommand\thefigure{A\arabic{figure}}    
\setcounter{figure}{0}   

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/power_lm.pdf}
\caption{{\bf Power to detect a statistically significant positive correlation between infection prevalence and frequency of sexual hosts.}
Pearson correlation was used to test for correlation between square root arcsine transformed infection prevalence and frequency of sexual hosts in simulated data from the posterior distributions.
}
\label{fig:power_lm}
\end{figure}

\end{document}
