\documentclass{article}
\title{Quantifying the effects of parasites on the maintenance of sex}
\author{Sang Woo Park, Benjamin M Bolker}

\usepackage{tabularx}

\usepackage{amsmath}
\usepackage{natbib}
\usepackage{hyperref}
\bibliographystyle{chicago}
\date{\today}

\newcommand{\etal}{\textit{et al.}}

\newcommand{\fref}[1]{Fig.~\ref{fig:#1}}

\begin{document}

\maketitle

\section{Introduction}

Sexual reproduction is followed by numerous costs \citep{lehtonen2012many},
and these costs are often attributed to cost of producing males \citep{smith1978evolution}.
As males cannot produce offspring, a sexually reproducing population should be outgrown by an asexually reproducing population growing twice as fast (given that the sexual population produces 50\% males and 50\% females) --
hence the infamous \emph{two-fold cost of sex} \citep{smith1978evolution}.
However, the two-fold cost of sex relies on the assumption that all else is equal, and this is not necessarily true in nature.
In fact, sexual reproduction is much more common in nature \citep{vrijenhoek1998animal}.
So what drives sex to persist?

One explanation for sex is the Red Queen Hypothesis \citep{bell1982masterpiece}.
The Red Queen Hypothesis predicts sexually reproducing organisms to overcome the two-fold cost of sex under strong parasite selection by producing genetically diverse offspring that are resistant to infection \citep{jbs1949disease, jaenike1978hypothesis, hamilton1980sex}.
Hosts with rare genotypes are expected to result in higher fitness and become common eventually.
Furthermore, parasites coevolves with the host population to infect the common "rare" genotype [CITE],
and negative frequeny dependnece drives an oscillation in genotypic frequencies in host and parasite populations \citep{clarke1976ecological, hamilton1980sex}.

Since the Red Queen Hypothesis has been postulated, much of theoretical work has focused on figuring out conditions under which parasites can maintain sexual reproduction.

On the other hand, empiricists have tried to quantify the Red Queen.

While the Red Queen Hypothesis is well supported both empirically and theoretically, there still remains a gap between theory and data.
Many models for Red Queen Hypothesis rely on simplifying assumption that are not applicable to natural populations \citep{ashby2015diversity}.
For example, 

Recent studies have shown that ecological feedbacks can allow sex to be maintained more easily [CITE].
We find that a simple model can capture the dynamics 
Here, we try to bridge this gap further.

% \cite{lively2001trematode} successfully applied ???. Using a mathematical model, he predicted that prevalence of infection should be positively correlated with prevalence of sex given large variaton among population \citep{lively2001trematode} and was able to confirm empirically [CITE]. However, such correlation was not observed in a similar snail-trematode system [CITE] and Lively's original model rather relies on simple assumption.

In this study, we extend the model suggested by \cite{lively2010epidemiological} to include stochasticity and population structure and fit the to data from \cite{dagan2013clonal, mckone2016fine, vergara2014infection} using approximate bayesian computation (ABC).
[Something about what we found out from the model fitting]
Using posterior distributions obtained from the fit, we perform power analysis to test the idea that infection prevalence is positively correlated with frequency of sexual reproduction \cite{lively2001trematode}. 
We find that cycling of parasites and hosts can lead to negative correlation between prevalence of infection and sex. Instead, positive correlated predicted by \cite{lively2001trematode} is driven by populations that do not go through Red Queen dynamics (i.e., having low level of sexual and infection).

\section{Methods}

\subsection{Data}

We consider two snail-trematode population in New Zealand \citep{mckone2016fine, vergara2014infection} and a similar snail-trematode population in Isral \citep{dagan2013clonal}. First, the snail-trematode system in New Zealand has been extensively studied under the Red Queen Hypothesis [CITE] so we expect a Red Queen model to fit well. Second, [Something about Israel snails being different]. Finally, by considering both spatial \citep{dagan2013clonal, mckone2016fine} and spatiotemporal data \citep{vergara2014infection}, we can compare how structure of the data may affect model fits. 
Data for \cite{vergara2014infection} and \cite{dagan2013clonal} were obtained from their Dryad repository [CITE] and data for \cite{mckone2016fine} was extracted from their figure.

\subsection{Model}

We model obligate sexual hosts competing with obligate asexual hosts in multiple population by extending the model introduced by \cite{lively2010epidemiological}.
A model that \cite{lively2010epidemiological} used is essentially an SI model on a discrete time scale with natural mortality and virulence (reduction in offspring production by infected hosts).
It is a suitable canditate for this study as it captures essential structures that are present in basic epidemiological and population population dynamics models and is general enough to be applied to broad systems in nature.
We do not consider mechanistic details of the snail-trematode system such as life history of trematodes \citep{vergara2014infection} as it would require us to estimate or make assumptions on parameters on mechanisms, both of which are unrealistic.

All hosts are assumed to be diploids with two biallelic loci, and parasites are assumed to be haploids.
Let $S_{ij}^k(t)$ and $A_{ij}^k(t)$ be the number of sexual and asexual hosts with genotype $ij$ from population $k$ at generation $t$. 
For simplicity, we drop the superscript representing population and write $S_{ij}(t)$ and $A_{ij}(t)$ unless necessary.
Following \cite{lively2010epidemiological}, the expected number of offsprings (without recombination or outcrossing) produced by a sexual population is given by
\begin{equation}
S_{ij}' = c_b (1-s) \left(W_U S_{ij,U} (t) + W_I S_{ij,I} (t)\right),
\end{equation}
where $s$ is the proportion of males produced, $S_{ij, U}$ and $S_{ij,I}$ are the number of uninfected and infected sexual hosts in a population, and $W_U$ and $W_I$ are their corresponding fitnesses.
We allow for cost of sex to vary by multiplying $c_b$ to the growth term ($2/c_b$ corresponds to two fold cost of sex) \citep{ashby2015diversity}.
Equivalently, $S_{ij}'$ is the amount of genotypic contribution by sexual hosts to the next generation.

Asexual hosts are assumed to be strictly clonal. Then, the expected number of offsprings produced by an asexual population is given by
\begin{equation}
A_{ij}' = W_U A_{ij,U} (t) + W_I A_{ij,I} (t),
\end{equation}
where $A_{ij, U}$ and $A_{ij,I}$ are the number of uninfected and infected asexual hosts in a population.

We assume that there is mixing between population, determined by rate $\epsilon_{\textrm{mix}}$.
Then, the expected number of offspring in the next generation (accounting for contributions from all populations) is given by
\begin{equation}
\begin{aligned}
\mathrm{E}\left(S_{ij}^k(t+1)\right) &= \mathrm{Sex} \left((1 - \epsilon_{\textrm{mix}}) \left(S_{ij}^k\right)' + \frac{\epsilon_{\textrm{mix}}}{n_{\textrm{population}}-1} \sum_{l \neq k} \left(S_{ij}^l\right)'\right),\\
\mathrm{E}\left(A_{ij}^k(t+1)\right) &= (1 - \epsilon_{\textrm{mix}}) \left(A_{ij}^k\right)' + \frac{\epsilon_{\textrm{mix}}}{n_{\textrm{population}}-1} \sum_{l \neq k} \left(A_{ij}^l\right)',\\
\end{aligned}
\end{equation}
where $\mathrm{Sex}$ is a function defined by recombination at a rate $r_{\textrm{host}}$ and outcrossing, and $n_{\textrm{population}}$ is the number of populations present in the system.
We then take a poisson random variable to simulate process error and allow for stochastic migration to avoid fixation:
\begin{equation}
\begin{aligned}
S_{ij}^k(t+1) &\sim \mathrm{Poisson}\left(\lambda=\mathrm{E}\left(S_{ij}^k(t+1)\right)\right) + \mathrm{Bernoulli}\left(p=p_{ij,\textrm{sex}}\right),\\
A_{ij}^k(t+1) &\sim \mathrm{Poisson}\left(\lambda=\mathrm{E}\left(A_{ij}^k(t+1)\right)\right) + \mathrm{Bernoulli}\left(p=p_{ij, \textrm{asex}}\right),
\end{aligned}
\end{equation}
where $p_{ij,\textrm{sex}}$ and $p_{ij, \textrm{asex}}$ are the probability of a sexual and an asexual host with genotype $ij$ entering the popualation.

Infection is modeled using the matching alleles model \citep{otto1998evolution}.
We assume that heterozygous individuals are equally susceptible to parasites that match either genotypes 

The expected number of infected hosts that carry parasite with genotype $i$ at generation $t$ is given by:
\begin{equation}
I_{i}(t) = \sum_{p} 2^{\delta_{ij}} \left( S_{ip,i,I}(t) + A_{ip,i,I}(t)\right),
\end{equation}
where $S_{ip,i,I}$ and $A_{ip,i,I}$ are the expected number of sexual and asexual hosts infected with genotype $i$ parasite. 
Following \cite{ashby2015diversity}, we assume that mutation can occur in one locus with probability $r_{\textrm{parasite}}$. 
We also allow for external migration of an infected host carrying parasite $i$ with probability $p_{i, \textrm{parasite}}$ to avoid fixation.

The amount of infectious contacts made by infected hosts from a population is given by $\lambda_i^k = \beta^k {I_i'}^k(t)$, where $\beta^K$ is the transmission rate at each population, and $I_i'(t)$ is the number of infected hosts accounting for mutation. 
Since we allow for mixing between population, infected hosts in a population can make contact to susceptible hosts in other populations as well.
Then, all infectious contacts made by infected hosts that applies to a host in a single population is
\begin{equation}
\lambda_{i, \textrm{total}} = (1 - \epsilon_{\textrm{site}}) \lambda_i + \frac{\epsilon_{\textrm{site}}}{n_\textrm{site}-1} \sum_{l \neq k} \lambda_i^l
\end{equation}
Then, the force of infection that a susceptible host experiences is
\begin{equation}
\mathrm{FOI}_{ij}^k = \frac{\lambda_{i, \textrm{total}}  + \lambda_{j, \textrm{total}}  }{2 N(t+1)}
\end{equation}
The probability of infection for a host with genotype $ij$ at site $k$ in the next generation is
\begin{equation}
P_{ij}^k(t+1) = 1 - \exp\left(\mathrm{FOI}_{ij}^k\right).
\end{equation}

Finally, number of infected hosts in the next generation is determined by a binomial process:
\begin{equation}
\begin{aligned}
S_{ij,I} (t+1) &\sim \mathrm{Binom}(S_{ij} (t+1), P_{ij}),\\
A_{ij,I} (t+1) &\sim \mathrm{Binom}(A_{ij} (t+1), P_{ij}).
\end{aligned}
\end{equation}
Expected number of individuals infected with haplotype $i$ in the next generation is given by a ratio of $\lambda$:
\begin{equation}
\begin{aligned}
S_{ij,i,I}(t+1) &= \frac{\lambda_{i, \textrm{total}}}{\lambda_{i, \textrm{total}} + \lambda_{j, \textrm{total}}} S_{ij,I}(t+1)\\
A_{ij,i,I}(t+1) &= \frac{\lambda_{i, \textrm{total}}}{\lambda_{i, \textrm{total}} + \lambda_{j, \textrm{total}}} A_{ij,I}(t+1)
\end{aligned}
\end{equation}

\subsection{Simulation design and parameterization}

Most Red Queen models have studied a competition between a single asexual genotype and sexual genotypes or have assumed equal genetic diversity of asexual and sexual hosts (see \citep{ashby2015diversity}) but neither of these assumptions are realistic.
Instead, \cite{ashby2015diversity} adopted a more realistic approach by allowing for stochastic migration of an asexual genotype to a population.
Here, we combine these methods.
We allow for stochastic migration of migration of asexual hosts with genotypes genotypes but fix number of asexual genotypes (denoted by $G_{\textrm{asex}}$) that can be present in the population, 
while the number of sexual genotypes ($G_{\textrm{sex}}$) that can be present in the population is equal to size of the genotypic space ($=10$ in diploid hosts with two biallelic loci).

%% justification here or in the discussion?

In the beginning of every simlation, $G_{\textrm{asex}}$ is randomly chosen between 1 and 10 and asexual genotypes that can be introduced to the population are uniformly sampled from the entire genotypic space.
To account for differing number of sexual and asexual genotypes, we let 
\begin{equation}
\begin{aligned}
p_{ij, \textrm{sex}} &= 1-(1-p_{\textrm{host}})^{1/G_\textrm{sex}},\\
p_{ij, \textrm{asex}} &=
\begin{cases}
1-(1-p_{\textrm{host}})^{1/G_\textrm{asex}} & \text{if } ij \in \{\text{asexual genotypes}\} \\
0 & \text{otherwise}
\end{cases},
\end{aligned}
\end{equation}
where $p_{\textrm{host}}$ is the probability that at least one sexual and asexual host enters the population in a generation. We scale the probability of infected host carrying parsite genotype $i$ in a similar way for interpretability:
\begin{equation}
p_{i, \textrm{parasite}} = 1 - (1-p_{\textrm{infected}})^{1/4},
\end{equation}
where $p_{\textrm{infected}}$ is the probability that at least one infected host enters the population in a generation.

Each simulation consists of 40 populations. Every population consists of 2000 sexual hosts where 80 of them are infected. They are assumed to be in Hardy-Weinberg equilibrium with ratio between alleles being exactly half. Simulation runs for 500 generations without introduction of asexuals. At generation 501, 10 asexual hosts of a single genotype are introduced to each population (note that asexual genotype introduced can vary across population) and simulation runs for 600 generations while allowing for stochastic migration of asexuals as well.

\subsection{Approximate Bayesian Computation}

We use Approximate Bayesian Computation (ABC) to fit the model \citep{toni2009approximate}.
ABC relies on comparing summary statistics of observed data and those of simulated data and can be particularly useful when the exact likelihood function is not available [CITE?].
We consider mean proportion of infected and sexually reproducing snails in the system and variation in these proportions -- measured by coefficient of variation (CV) -- across space and time as summary statistics of interest.

For each sample from the posterior, the model is simulated and a subsample from the simulated data is drawn so that the number of simulated population is equal to the number of sites collected in a study.
Then, the summary statistics are calculated by considering last 100 generations of simulated data and the parameter sample as well as the simulation generated by the sample is accepted if the distance between simulated and observed data (measureed by the absolute sum of differences in summary statistics) is less than the tolerance value.
By simulating greather number of populations than the observed, we account for uncertainty in unobserved population as well as interaction between populations.

We use uninformative priors for all parameters except $c_b$, a scale parameter for the cost of sex.
The prior distribution for the scale parameter is chosen so that 95\% quantile of cost of sex (2/$c_b$) is approximately equal to the 95\% confidence interval reported by \cite{gibson2017two} (see ??? for the list of parameters).
To allow for more efficient estimation of the posterior distribution, we use Sequential Monte Carlo \cite{turner2012tutorial} (see Appendix).

\section{Results}

\subsection{Model fit}

Before we present the parameter estimates, we first discuss the resulting fits of the model.
When we compare the summary statistics of the observed data and those estimated by the model, it appears that the simple Red Queen model can reproduce observed data reasonably well (\fref{smcsumm}).
The only discrepancy is that the model overestimates prevalence of infection, given similar level of sexual reproduction.
In particular, the data presented by \cite{mckone2016fine} has mean infection prevalence of approximately 5\% but the model predicts prevalence of approximately [TODO].
It's possible that running ABC with lower tolerance can reduce the difference but the discrepancy we see here is most likely caused by the model.

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/smc_summary.pdf}
\caption{{\bf Need caption.}
Need caption}
\label{fig:smcsumm}
\end{figure}

High estimate of infection prevalence leads us to the the most obvious limitation of the model: genetic diversity. 
Moderately high genetic diversity can allow sexual hosts to escape infection more easily and therefore reduces prevalence of infection while maintaining sexual reproduction \citep{lively2010effect, king2012does, ashby2015diversity}.
However, our model, and many other models that study the Red Queen Hypothesis [CITE], assume that infection is determined by two biallelic loci. 
Although not much is known about genes that determine trematode infecion, it is not likely that a two loci model is sufficient to capture the genetic diversity of snails [CITE].

As summary statistics only show marginal fits of the model, we also compared mean infection prevalence and mean proportion of sexually reproducing hosts of each simulated population against observed data (\fref{ivs}).
\fref{ivs} may seem to suggest the model fits are much worse than what we expected in \fref{smcsumm}.
However, we are plotting the mean proportions (suppressing the effects of the cycle) whereas observed data is more likely to be crosssections of the Red Queen cycles.
It is still surprising that the model predicts similar summary statistics as data observed by \cite{dagan2013clonal} but show a completely different trend when we compare the trends. 
Whereas our model predicts sexual reproduction to be maintained at intermediate prevalence of infection, two truly mixed cites from \cite{dagan2013clonal} have much lower prevalence and any sites with higher prevalence have almost no sexually reproducing snails.

The most striking result from \fref{ivs} is the negative correlation between prevalence of infection and frequency of sexually reproducing hosts in the intermediate range of prevalence (most clearly seen from the fit to \cite{vergara2014infection}).
As 


\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/simulated_data.pdf}
\caption{{\bf Need caption.}
Need caption}
\label{fig:ivs}
\end{figure}

\subsection{Power analysis}

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/true_effect.pdf}
\caption{{\bf Need caption.}
Need caption}
\label{fig:teff}
\end{figure}


\subsection{Parameter estimation}

Our model estimates wide ranges of parameters across data sets (\fref{smcparam}).
The most surprising result is the wide posterior of the scale parameter, $c_b$, for cost of sex (\fref{smcparam}). 
\cite{ashby2015diversity} defined $c_b$ as additional costs and benefits of sex, where $c_b=1$ corresponds the two fold cost.
Under their interpretation, our estimate corresponds to the following 95\% credible intervals for cost of sex: [???].
However, the meaning of the parameter changes from an inferential point of view.

\begin{figure}[!ht]
\includegraphics[width=\textwidth]{../fig/smc_param.pdf}
\caption{{\bf Need caption.}
Need caption}
\label{fig:smcparam}
\end{figure}

In the absence of the parameter $c_b$ (i.e, when $c_b$ = 1), our model assumes two fold cost of sex in growth rate.
Then, an estimate of $c_b > 1$ can be interpreted as the amount of additional compensation required in order to overcome the two fold cost of sex.
Additional compensation may imply other mechanisms that maintains, reduced cost of sex, or higher genetic diversity, a point to which we will come back later.
Likewise, an estimate of $c_b < 1$ implies that the model can sufficiently produce observed data.

We also find that the model estimates high virulence overall and almost 100\% virulence for the system observed by \cite{mckone2016fine}.



\section{Discussion}






\bibliography{redqueen}
\end{document}
