\documentclass{article}
\title{Quantifying the effects of parasites on the maintenance of sex}
\author{Sang Woo Park, Benjamin M Bolker}

\usepackage{tabularx}

\usepackage{amsmath}
\usepackage{natbib}
\usepackage{hyperref}
\bibliographystyle{chicago}
\date{\today}

\newcommand{\etal}{\textit{et al.}}

\newcommand{\fref}[1]{Fig.~\ref{fig:#1}}

\begin{document}

\maketitle

\section{Introduction}

The evolution of sexual reproduction still remains largely unsolved [CITE].
Despite being the dominant mode of reproduction \citep{vrijenhoek1998animal}, sexual reproduction is followed by many costs \citep{lehtonen2012many}.
The most often cited is perhaps the cost of producing males \citep{smith1978evolution}.
As males cannot produce offspring, a sexually reproducing population is expected to be outgrown by an asexually reproducing counterpart that grows twice as fast (given that the sexual population produces 50\% males and 50\% females);
hence, cost of sex is often referred to as the infamous \emph{two-fold cost of sex} \citep{smith1978evolution}.
However, the two-fold cost of sex relies on the assumption that all else is equal, and this is not necessarily true in nature.
So what drives sex to persist?

One explanation for sex is the Red Queen Hypothesis \citep{bell1982masterpiece}.
The Red Queen Hypothesis predicts sexually reproducing organisms to overcome the two-fold cost of sex under strong parasite selection by producing genetically diverse offspring that are resistant to infection \citep{jbs1949disease, jaenike1978hypothesis, hamilton1980sex}.
Hosts with rare genotypes are expected to result in higher fitness and become common eventually.
Furthermore, parasites coevolves with the host population to infect the common "rare" genotype [CITE],
and negative frequeny dependnece drives an oscillation in genotypic frequencies in host and parasite populations \citep{clarke1976ecological, hamilton1980sex}.

Since the Red Queen Hypothesis has been postulated, much of theoretical work has focused on figuring out conditions under which parasites can maintain sexual reproduction.

On the other hand, empiricists have tried to quantify the Red Queen.

While the Red Queen Hypothesis is well supported both empirically and theoretically, there still remains a gap between theory and data.
Many models for Red Queen Hypothesis rely on simplifying assumption that are not applicable to natural populations \citep{ashby2015diversity}.
For example, 

Recent studies have shown that ecological feedbacks can allow sex to be maintained more easily [CITE].
We find that a simple model can capture the dynamics 
Here, we try to bridge this gap further.

% \cite{lively2001trematode} successfully applied ???. Using a mathematical model, he predicted that prevalence of infection should be positively correlated with prevalence of sex given large variaton among population \citep{lively2001trematode} and was able to confirm empirically [CITE]. However, such correlation was not observed in a similar snail-trematode system [CITE] and Lively's original model rather relies on simple assumption.

In this study, we extend the model suggested by \cite{lively2010epidemiological} to include stochasticity and population structure and fit the to data from \cite{dagan2013clonal, mckone2016fine, vergara2014infection} using approximate bayesian computation (ABC).
[Something about what we found out from the model fitting]
Using posterior distributions obtained from the fit, we perform power analysis to test the idea that infection prevalence is positively correlated with frequency of sexual reproduction \cite{lively2001trematode}. 
We find that cycling of parasites and hosts can lead to negative correlation between prevalence of infection and sex. Instead, positive correlated predicted by \cite{lively2001trematode} is driven by populations that do not go through Red Queen dynamics (i.e., having low level of sexual and infection).

\section{Methods}

\subsection{Data}

We consider two snail-trematode population in New Zealand \citep{mckone2016fine, vergara2014infection} and a similar snail-trematode population in Isral \citep{dagan2013clonal}. First, the snail-trematode system in New Zealand has been extensively studied under the Red Queen Hypothesis [CITE] so we expect a Red Queen model to fit well. Second, [Something about Israel snails being different]. Finally, by considering both spatial \citep{dagan2013clonal, mckone2016fine} and spatiotemporal data \citep{vergara2014infection}, we can compare how structure of the data may affect model fits. 
Data for \cite{vergara2014infection} and \cite{dagan2013clonal} were obtained from their Dryad repository [CITE] and data for \cite{mckone2016fine} was extracted from their figure.

\subsection{Model}

We model obligate sexual hosts competing with obligate asexual hosts in multiple population by extending the model introduced by \cite{lively2010epidemiological}.
A model that \cite{lively2010epidemiological} used is essentially an SI model on a discrete time scale with natural mortality and virulence (reduction in offspring production by infected hosts).
It is a suitable canditate for this study as it captures essential structures that are present in basic epidemiological and population population dynamics models and is general enough to be applied to broad systems in nature.
We do not consider mechanistic details of the snail-trematode system such as life history of trematodes \citep{vergara2014infection} as it would require us to estimate or make assumptions on parameters on mechanisms, both of which are unrealistic.

All hosts are assumed to be diploids with two biallelic loci, and parasites are assumed to be haploids.
Let $S_{ij}^k(t)$ and $A_{ij}^k(t)$ be the number of sexual and asexual hosts with genotype $ij$ from population $k$ at generation $t$. 
For simplicity, we drop the superscript representing population and write $S_{ij}(t)$ and $A_{ij}(t)$ unless necessary.
Following \cite{lively2010epidemiological}, the expected number of offsprings (without recombination or outcrossing) produced by a sexual population is given by
\begin{equation}
S_{ij}' = c_b (1-s) \left(W_U S_{ij,U} (t) + W_I S_{ij,I} (t)\right),
\end{equation}
where $s$ is the proportion of males produced, $S_{ij, U}$ and $S_{ij,I}$ are the number of uninfected and infected sexual hosts in a population, and $W_U$ and $W_I$ are their corresponding fitnesses.
We allow for cost of sex to vary by using a scale parameter, $c_b$, to the growth term ($2/c_b$ corresponds to two fold cost of sex) \citep{ashby2015diversity}.
Equivalently, $S_{ij}'$ is the amount of genotypic contribution by sexual hosts to the next generation.

Asexual hosts are assumed to be strictly clonal. Then, the expected number of offsprings produced by an asexual population is given by
\begin{equation}
A_{ij}' = W_U A_{ij,U} (t) + W_I A_{ij,I} (t),
\end{equation}
where $A_{ij, U}$ and $A_{ij,I}$ are the number of uninfected and infected asexual hosts in a population.

We assume that there is mixing between population, determined by rate $\epsilon_{\textrm{mix}}$.
Then, the expected number of offspring in the next generation (accounting for contributions from all populations) is given by
\begin{equation}
\begin{aligned}
\mathrm{E}\left(S_{ij}^k(t+1)\right) &= f_{\textrm{sex}}\left((1 - \epsilon_{\textrm{mix}}) \left(S_{ij}^k\right)' + \frac{\epsilon_{\textrm{mix}}}{n_{\textrm{population}}-1} \sum_{l \neq k} \left(S_{ij}^l\right)'\right),\\
\mathrm{E}\left(A_{ij}^k(t+1)\right) &= (1 - \epsilon_{\textrm{mix}}) \left(A_{ij}^k\right)' + \frac{\epsilon_{\textrm{mix}}}{n_{\textrm{population}}-1} \sum_{l \neq k} \left(A_{ij}^l\right)',\\
\end{aligned}
\end{equation}
where $f_{\textrm{sex}}(x)$ is the function that models sexual reproduction, including recombination at a rate $r_{\textrm{host}}$ and outcrossing, and $n_{\textrm{population}}$ is the number of populations present in the system.
We then take a poisson random variable to simulate process error and allow for stochastic migration to avoid fixation:
\begin{equation}
\begin{aligned}
S_{ij}^k(t+1) &\sim \mathrm{Poisson}\left(\lambda=\mathrm{E}\left(S_{ij}^k(t+1)\right)\right) + \mathrm{Bernoulli}\left(p=p_{ij,\textrm{sex}}\right),\\
A_{ij}^k(t+1) &\sim \mathrm{Poisson}\left(\lambda=\mathrm{E}\left(A_{ij}^k(t+1)\right)\right) + \mathrm{Bernoulli}\left(p=p_{ij, \textrm{asex}}\right),
\end{aligned}
\end{equation}
where $p_{ij,\textrm{sex}}$ and $p_{ij, \textrm{asex}}$ are the probability of a sexual and an asexual host with genotype $ij$ entering the popualation.

Infection is modeled using the matching alleles model \citep{otto1998evolution}.
We assume that heterozygous individuals are equally susceptible to parasites that match either genotypes 

The expected number of infected hosts that carry parasite with genotype $i$ at generation $t$ is given by:
\begin{equation}
I_{i}(t) = \sum_{p} 2^{\delta_{ij}} \left( S_{ip,i,I}(t) + A_{ip,i,I}(t)\right),
\end{equation}
where $S_{ip,i,I}$ and $A_{ip,i,I}$ are the expected number of sexual and asexual hosts infected with genotype $i$ parasite. 
Following \cite{ashby2015diversity}, we assume that mutation can occur in one locus with probability $r_{\textrm{parasite}}$. 
We also allow for external migration of an infected host carrying parasite $i$ with probability $p_{i, \textrm{parasite}}$ to avoid fixation.

The amount of infectious contacts made by infected hosts from a population is given by $\lambda_i^k = \beta^k {I_i'}^k(t)$, where $\beta^K$ is the transmission rate at each population, and $I_i'(t)$ is the number of infected hosts accounting for mutation. 
Since we allow for mixing between population, infected hosts in a population can make contact to susceptible hosts in other populations as well.
Then, all infectious contacts made by infected hosts that applies to a host in a single population is
\begin{equation}
\lambda_{i, \textrm{total}} = (1 - \epsilon_{\textrm{site}}) \lambda_i + \frac{\epsilon_{\textrm{site}}}{n_\textrm{site}-1} \sum_{l \neq k} \lambda_i^l
\end{equation}
Then, the force of infection that a susceptible host experiences is
\begin{equation}
\mathrm{FOI}_{ij}^k = \frac{\lambda_{i, \textrm{total}}  + \lambda_{j, \textrm{total}}  }{2 N(t+1)}
\end{equation}
The probability of infection for a host with genotype $ij$ at site $k$ in the next generation is
\begin{equation}
P_{ij}^k(t+1) = 1 - \exp\left(\mathrm{FOI}_{ij}^k\right).
\end{equation}

Finally, number of infected hosts in the next generation is determined by a binomial process:
\begin{equation}
\begin{aligned}
S_{ij,I} (t+1) &\sim \mathrm{Binom}(S_{ij} (t+1), P_{ij}),\\
A_{ij,I} (t+1) &\sim \mathrm{Binom}(A_{ij} (t+1), P_{ij}).
\end{aligned}
\end{equation}
Expected number of individuals infected with haplotype $i$ in the next generation is given by a ratio of $\lambda$:
\begin{equation}
\begin{aligned}
S_{ij,i,I}(t+1) &= \frac{\lambda_{i, \textrm{total}}}{\lambda_{i, \textrm{total}} + \lambda_{j, \textrm{total}}} S_{ij,I}(t+1)\\
A_{ij,i,I}(t+1) &= \frac{\lambda_{i, \textrm{total}}}{\lambda_{i, \textrm{total}} + \lambda_{j, \textrm{total}}} A_{ij,I}(t+1)
\end{aligned}
\end{equation}

\subsection{Simulation design and parameterization}

Most Red Queen models have studied a competition between a single asexual genotype and sexual genotypes or have assumed equal genetic diversity of asexual and sexual hosts (see \citep{ashby2015diversity}) but neither of these assumptions are realistic.
Instead, \cite{ashby2015diversity} adopted a more realistic approach by allowing for stochastic migration of an asexual genotype to a population.
Here, we combine these methods.
We allow for stochastic migration of migration of asexual hosts with genotypes genotypes but fix number of asexual genotypes (denoted by $G_{\textrm{asex}}$) that can be present in the population, 
while the number of sexual genotypes ($G_{\textrm{sex}}$) that can be present in the population is equal to size of the genotypic space ($=10$ in diploid hosts with two biallelic loci).

In the beginning of every simlation, $G_{\textrm{asex}}$ is randomly chosen between 1 and 10 and asexual genotypes that can be introduced to the population are uniformly sampled from the entire genotypic space.
To account for differing number of sexual and asexual genotypes, we let 
\begin{equation}
\begin{aligned}
p_{ij, \textrm{sex}} &= 1-(1-p_{\textrm{host}})^{1/G_\textrm{sex}},\\
p_{ij, \textrm{asex}} &=
\begin{cases}
1-(1-p_{\textrm{host}})^{1/G_\textrm{asex}} & \text{if } ij \in \{\text{asexual genotypes}\} \\
0 & \text{otherwise}
\end{cases},
\end{aligned}
\end{equation}
where $p_{\textrm{host}}$ is the probability that at least one sexual and asexual host enters the population in a generation. We scale the probability of infected host carrying parsite genotype $i$ in a similar way for interpretability:
\begin{equation}
p_{i, \textrm{parasite}} = 1 - (1-p_{\textrm{infected}})^{1/4},
\end{equation}
where $p_{\textrm{infected}}$ is the probability that at least one infected host enters the population in a generation.

Each simulation consists of 40 populations. Every population consists of 2000 sexual hosts where 80 of them are infected. They are assumed to be in Hardy-Weinberg equilibrium with ratio between alleles being exactly half. Transmission rate parameter, $\beta$, is randomly drawn from a lognormal distribution with parameters $\beta_{\textrm{meanlog}}$ and $\beta_{\textrm{sdlog}}$ for each population. Simulation runs for 500 generations without introduction of asexuals. At generation 501, 10 asexual hosts of a single genotype are introduced to each population (note that asexual genotype introduced can vary across population) and simulation runs for 600 generations while allowing for stochastic migration of asexuals as well.

\subsection{Approximate Bayesian Computation}

We use Approximate Bayesian Computation (ABC) to fit the model \citep{toni2009approximate}.
ABC relies on comparing summary statistics of observed data and those of simulated data and can be particularly useful when the exact likelihood function is not available [CITE?].
We consider mean proportion of infected and sexually reproducing snails in the system and variation in these proportions -- measured by coefficient of variation (CV) -- across space and time as summary statistics of interest.

For each sample from the posterior, the model is simulated and a subsample from the simulated data is drawn so that the number of simulated population is equal to the number of sites collected in a study.
Then, the summary statistics are calculated by considering last 100 generations of simulated data and the parameter sample as well as the simulation generated by the sample is accepted if the distance between simulated and observed data (measureed by the absolute sum of differences in summary statistics) is less than the tolerance value.
By simulating greater number of populations than number of sampled sites, we account for uncertainty in unobserved population as well as interaction between populations.

We use uninformative priors for all parameters except $c_b$, a scale parameter for the cost of sex.
The prior distribution for the scale parameter is chosen so that 95\% quantile of cost of sex (2/$c_b$) is approximately equal to the 95\% confidence interval reported by \cite{gibson2017two} (see ??? for the list of parameters).
As \cite{dagan2013clonal} and \cite{mckone2016fine} report proportion of males, instead of proportion of sexual hosts, we double the proportion of males reported when we fit the model.

To allow for more efficient estimation of the posterior distribution, we use Sequential Monte Carlo \cite{turner2012tutorial} (see Appendix for details).

\subsection{Power analysis}

To calculate the power to detect positive correlation between infection prevalence and frequency of sexual hosts, we simulate an observed sample by drawing a random multinomial sample from simulated data.
For simplicity, we assume that a year contains of two snail generations [CITE] and that samples are taken within a short period of time.
For each simulation from the posterior distributions, we start by setting a reference generation to generatation 1001 and choosing $n$ populations at random from 40 simulated popoulations.
For each selected population, hosts are divided into four categories based on their infection status (infected/uninfected) and reproductive mode (asexual/sexual),
and mean proportion of hosts in each category is calculated by averaging over two consecutive generations.
A multinomial sample of size $m$ is drawn from each selected population independently based on the four categories. 
Positive correlation between proportion of infected hosts and proportion of sexual hosts is tested using Spearman's rank correlation at 5\% significance level.
We repeat the process 100 times for each simulation from a posterior distribution.

\section{Results}

Our model can reproduce infection prevalence and frequency of sexual hosts observed acrosss three snail-trematode systems reasonably well (\fref{smcsumm}).
However, our model tends to overestimate mean infection prevalence across all three systems.
In particular, \cite{mckone2016fine} reported mean infection prevalence of 5.1\% but the mean (95\% quantile) estimate of the fitted model is 23.7\% (9.1\% - 40.3\%).
The fitted model underestimates mean frequency of sexual hosts in \cite{vergara2014infection} population by almost 10\% when compared with the mean estimate but the observed mean frequency still lies within a reasonable range of estimated values.
We also find that there is large variation in the estimates even though model fitting was performed based on these summary statistics.

\begin{figure}[htbp]
\includegraphics[width=\textwidth]{../fig/smc_summary.pdf}
\caption{{\bf Summary statistics of the observed data v.s. summary statistics of the simulated data from the posterior samples.}
Violin plots show distribution of summary statistics from each simulated set. For each simulated data set accepted along with a posterior parameter sample, simulated populations are sampled at random 100 times so that each sample consists of equal number of populations as number of sites in fitted data. Then, summary statistics are calculated for each sample by taking last 100 generations. Squares represnt observed summary statistics.
}
\label{fig:smcsumm}
\end{figure}

To further diagnose the fit, we compare the relationships between infection prevalence and frequency of sexual hosts predicted by the model with those observed across three systems (\fref{ivs}).
Despite being able to reproduce summary statistics reported by \cite{dagan2013clonal} well, 
our model is unable to predict qualitative trend bewteen proportion of sexual hosts and proportion of infected hosts (\fref{ivs}; \cite{dagan2013clonal}).
While both simulated data and observed data mostly consist of asexual populations,
our model predicts some amount of sexual reproduction to be maintained when infection prevalence is greather than 20\%. 
On the other hand, the data provided by \cite{dagan2013clonal} shows that sexual reproduction is only maintained at low levels of infection prevalence and is barely maintained otherwise.
Due to such badness of fit, we exclude fits to \cite{dagan2013clonal} from further discussion (including parameter estimation and power analysis).

\begin{figure}[htbp]
\includegraphics[width=\textwidth]{../fig/simulated_data.pdf}
\caption{{\bf Comparison in trends of the observed data v.s. simulated data from the posterior samples.}
Solid circles represent a population within a simulation. Triangles represent observed data.}
\label{fig:ivs}
\end{figure}

The most striking result from \fref{ivs} is that there is a region in which proportion of infected hosts increases while proportion of sexual hosts increases (most clearly seen from the fits to \cite{mckone2016fine} and \cite{vergara2014infection}). %% TODO: add circles around it??
As infection rate increases, selection for sexual hosts increases but increasing number of resistant offsprings prevents further infection from happening and hence decreases prevalence of infection.
This region connects the overlap in prevalence between sexual and asexual populations noted by \cite{lively2001trematode}
We also find that proportion of sexual hosts starts to decrease when infection prevalence is very high.
Decrease in fitness of sexual hosts with increase in prevalence was predicted by \cite{ashby2015diversity} but can also be also observed in an earlier work by \cite{lively2010epidemiological} (not discussed in the paper).

The model estimates similar parameters for both snail-trematode systems studied by \cite{mckone2016fine} and \cite{vergara2014infection} (\fref{smcparam}).
The most surprising result is that the posterior distributions of the scale parameter, $c_b$, estimated by the model is much wider than the prior distribution and have noticeably higher mean: (mean (95\% credible interval)) 1.19 (0.73-1.79) and 1.29 (0.79-2.01) for \cite{mckone2016fine} and \cite{vergara2014infection}, repsectively. %% TODO: Do we need to consider HPD region? does it matter? 
\cite{ashby2015diversity} defined $c_b$ as additional costs and benefits of sex, where $c_b=1$ corresponds the two fold cost.
Under their interpretation, our estimate corresponds to the following mean and 95\% CIs for cost of sex: 1.68 (1.12-2.73) and 1.55 (1.00-2.52), respectively.

\begin{figure}[htbp]
\includegraphics[width=\textwidth]{../fig/smc_param.pdf}
\caption{{\bf Parameter estimates from Sequential Monte Carlo Approximate Bayesian Computation.}
(dashed lines) prior distribution (solid lines) posterior distribution. 100 posterior samples were obtained from SMC ABC. Prior distributions are specified in [TODO]
}
\label{fig:smcparam}
\end{figure}

The model estimates high virulence overall and almost 100\% virulence for the data from \cite{mckone2016fine}: (mean (95\% CI)) 88.8\% (75.0\%-99.2\%) and 95.0\% (73.4\%-100.0\%) for \cite{vergara2014infection} and \cite{mckone2016fine}, repsectively.
In contrast to \cite{lively2010epidemiological}, who modeled 1 asexual genotype competing with 9 sexual genotypes, our model estimate shows that higher asexual to sexual genotypic ratio can be supported ($G_{\textrm{asex}}$ panel in \fref{ivs}).
However, it is still necessary for sexual hosts to have higher genetic diversity than asexual hosts.

Finally, a power analysis show that the power for detecting a positive correlation between infection prevalence and frequency of sexual hosts is not very high (\fref{power}).
Given that \cite{mckone2016fine} collected about 100 snails from each of the 18 sites, they would have had 45\% power to detect the positive correlation.
Surprisingly, the power to detect positive correlation in the system studied by \cite{vergara2014infection} is extremely low (less than 10\%) and there is much higher (but still low) power to detect a negative correlation.
Increasing number of samples per site has small effect on power once the sample size greater than 50.
Increasing number of sites leads to greater increase in power but the power appears to saturate as number of sites increases.

\begin{figure}[htbp]
\includegraphics[width=\textwidth]{../fig/power.pdf}
\caption{{\bf Power to detect a significant correlation between infection prevalence and frequency of sexual hosts.}
Spearman's rank correlation was used to detect correlation between infection prevalence and frequency of sexual hosts in simulated data from the posterior distributions. Detecting the positive correlation predicted by \cite{lively2001trematode} is difficult. Increainsg number of samples and number of sites increases power but it is more efficient to increase number of sites.}
\label{fig:power}
\end{figure}

\section{Discussion}

Our results challenge how the Red Queen Hypothesis for sex has been studied.
Previous studies have relied on assumed parameters to understand how parasitism maintains sexual reproduction in host population [CITE].
Here, we show that (1) model parameters are estimable and (2) a simple Red Queen model with population structure can reproduce key summary statistics observed across three different snail populations (\fref{smcsumm}).
While our model is able to reprouce observed summary statistics, there still remain some discrepancies between model prediction and observed trends in the relationship between infection prevalence and frequency of sexual hosts (\fref{ivs}).
These discrepancies suggest that existing theoretical framework for the Red Queen Hypothesis may not be sufficient to explain sexual reproduction observed in nature.
We also find that detecting a positive correlation between infection prevalence and frequency of sexual hosts is difficult (\fref{power}) and hence not a good measure for measuring the effects of the Red Queen.

A model that does not fit well can carry more information than a model that fits well.
In order for our model to produce similar summary statistics as those observed in the snail-trematode systems, it requires the scale parameter ($c_b$) for the cost of sex to have mean greater than 1 (\fref{smcparam}).
In other words, a model that assumes two fold cost of sex ($c_b=1$) would not have fitted well.
However, $c_b > 1$ does not necessarily imply that cost of sex is less than two fold.
Instead, an estimate of $c_b > 1$ should be interpreted as the amount of compensation required in order for the model to reproduce observed data.
Additional compensation can include less than two fold cost of sex or any other mechanisms that promote sex.

A suitable candidate for model compensation is increase in host genetic diversity.
Although exact genetic architecture that determines trematode infecion in snails is not known, genetic diversity of snails that have been documented is far greater than what we have assumed (two biallelic loci model yields 10 genotypes) \citep{king2011parasites, dagan2013clonal}.
In addition, increasing genetic diversity of the model may resolve overestimation of infection prevalence (\fref{smcsumm}).
Previous studies have shown that moderately high genetic diversity can allow sexual hosts to escape infection more easily and therefore reduces prevalence of infection given equal amount of sexual reproduction \citep{lively2010effect, king2012does, ashby2015diversity}.
Until recently, many theoretical models have assumed limited genetic diversity.
Of the M models reviewed by \cite{ashby2015diversity} (Table 2), N out of M models... [TODO]
%% We suggest modelers to not limit genetic diversity and empricists to document genetic architecture that determine parasite infection???

While all three studies \citep{dagan2013clonal, mckone2016fine, vergara2014infection} considered similar snail-trematode systems, the snail populations studied by \cite{dagan2013clonal} live in intrinsically different environments.
Some populations are subject to seasonal flash floods, which can affect reproduction strategies \citep{ben2007temporal}.
As a result, our simple model that does not take any exogeneous factors into account predicts a qualitatively different trend (\fref{ivs}).
We conclude that the Red Queen Hypothesis alone cannot explain maintenance of sexual reproduction in this population.
Overall, our results indicate that more modeling effort is required to understand prevalence of sexual reproduction in nature.

Our power analysis (\fref{power}) appears to contradict the positive correlation predicted by \cite{lively1992parthenogenesis, lively2001trematode} and findings of many empirical studies that have confirmed the prediction \citep{lively1987evidence, lively2002temporal, kumpulainen2004parasites, vergara2013geographic, mckone2016fine}.
Since parasitism is the only mechanism that drives sexual reproduction in our model, we expected the power to be high.
However. the power analysis predicts almost no power for detecting a positive corrleation in the population studied by \cite{vergara2014infection} and higher but still low power for detecting a negative correlation, contradicting an earlier work by \cite{vergara2013geographic} that reported a positive correlation between between infection prevalence and male frequency in the same population.
The key premise behind the postive correlation predicted by \cite{lively2001trematode} is that there must be large variation in infection prevalence.
In particular, comparing \cite{vergara2013geographic} and \cite{vergara2014infection} shows that the positive correlation is driven by differences between three populations from different habitats:
populations that do not support the Red Queen Hypothesis (because they live in evolutionary coldspots) act as leverage points. 
In figure 3 of \cite{kumpulainen2004parasites}, there a clearer distinction between two groups of populations that can support the Red Queen Hypothesis (high infection and high sexual reproduction) and those that cannot support the Red Queen Hypothesis (low infection and low sexual reproduction).
% These observations are consistent with an earlier work by \cite{lively2002temporal} who noted that snail populations could be separated into two clusters with low and high male frequencies.
Since both data sets provided by \cite{vergara2014infection} and \cite{mckone2016fine} only consists of populations with high and low infection prevalence respectively, there is not enough variance in prevalence to observe the difference between two groups of populations.
Therefore, the model predicts low power for detecting the positive correlation.

% While predicting a positive correlation between prevalence of infection and frequency of sexual hosts, \cite{lively2001trematode} also noted that there is a region at which expected infection prevalence overlaps between sexual and asexual populations. 
% Within this region lies the association between increase in sexual reproduction and decrease in infection prevalence observed in \fref{ivs}.
On the other hand, negative correlation between prevalence of infection and frequency of sexual hosts can be explained by cycling of host and parasite populations.
A main component of the Red Queen Hypothesis is that negative frequency dependence drives oscillation in host reproductive strategy [CITE].
When temporal variation is taken into account, association between infection prevalence and frequency of sexual hosts can change depending on what phase each population is going through in its cycle (\fref{cycle}).
The negative correlation in the population does not contradict the positive correlation predicted by \cite{lively2001trematode} because their prediction did not take temporal variation into account.

\begin{figure}[htbp]
\includegraphics[width=\textwidth]{../fig/cycle_example.pdf}
\caption{{\bf Simulated data from the posterior distribution fitted to \cite{vergara2014infection}.}
A particular simulation is chosen from the posterior distributions to demonstrate that it is possible to predict opposite relationship when temporal variation is taken into account. (Top left) each point represents proportion of infected hosts and proportion of sexually reproducing hosts of each population at each generation. Last hundred generations are plotted. (Top right) each point represents mean proporiton of infected hosts and mean proportion of sexually reproducing hosts of each population averaged over the last hundred generations. Dashed lines represent least squares fit to all point. (Bottom) A typical host parasite cycle observed in a population from this simulated data. %% TODO: need to report parameter values?
}
\label{fig:cycle}
\end{figure}

Depressing power to detect positive correlation between infection prevalence and frequency of sexual hosts can be concerning.
However, as lack of variation explains low power, sampling sites with large variation in infection prevalence will increase power to detect the positive correlation.
In the Appendix, we show that doubling the log standard deviation of the transmission rates from the posterior distribution increases power. %% TODO: show this?
Alternatively, %% TODO: write about other tests

%% still thinking about conclusion

\bibliography{redqueen}
\end{document}
